---
title: "DissociationTest.Rmd"
author: "Rayna M Harris"
date: November 20, 2017
#output: html_document
output:
  md_document:
    variant: markdown_github
---

This R Markdown document will walk through the analysis of hippocampal tissue prepared with two different methods. The "homogenized" samples were collected by punch then homogenized in homogenization buffer from the Promega Maxwell kit. The "dissociated samples" were also collected similarily but the cells was dissociated after being punch and before being homogenized. 

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(colorRamps) # for a matlab like color scheme

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/DG_onlyhomodiss/')
```


```{r import data, echo=FALSE, message=FALSE}
setwd("~/Github/DissociationTest/bin")
samples <- read.csv("../data/sampleinfo2.csv")
samples$Slice <- as.factor(samples$Slice)
```


```{r wrangle kallisto counts, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
kallistoDirs = dir("../data/JA16444/")
kallistoDirs = kallistoDirs[!grepl("\\.(R|py|pl|sh|xlsx?|txt|tsv|csv|org|md|obo|png|jpg|pdf)$",
        kallistoDirs, ignore.case=TRUE)]
kallistoFiles = paste0(kallistoDirs, "/abundance.tsv")
names(kallistoFiles) = kallistoDirs

setwd('~/Github/DissociationTest/data/JA16444/')
if(file.exists(kallistoFiles))
  kallistoData = lapply(
    kallistoFiles,
    read.table,
    sep = "\t",
    row.names = 1,
    header = TRUE
)


## this for loop uses the reduce function to make two data frame with counts or tpm from all the samples
ids = Reduce(f=union, x=lapply(kallistoData, rownames))
if (all(sapply(kallistoData, function(x) {all(rownames(x)==ids)}))) {
    count = data.frame(
        id = ids,
        sapply(kallistoData, function(x) {x$est_counts}),
        check.names = FALSE,
        stringsAsFactors = FALSE
    )
    tpm = data.frame(
        id = ids,
        sapply(kallistoData, function(x) {x$tpm}),
        check.names = FALSE,
        stringsAsFactors = FALSE
    )
}
 
## make a dataframe with the parts of the gene id as columns
geneids <- count[c(1)] 
geneids$ENSMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 1)
geneids$ENSMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 2)
geneids$OTTMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 3)
geneids$OTTMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 4)
geneids$transcript <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 5)
geneids$gene <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 6)
geneids$length <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 7)
geneids$structure1 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 8)
geneids$structure2 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 9)
geneids$structure3 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 10)
geneids$transcript_lenght <- as.factor(paste(geneids$transcript, geneids$length, sep="_"))

## prep count data for gene level analysis with DESeq2 (as opposed to transcript level anlaysis)
countbygene <- full_join(geneids, count)
countbygene <- countbygene[-c(1:6,8:12)]   ## keep gene name and counts for samples)

## lengthen the dataframe, then wide with gene level sums, then make gene the row name, then round the value to nearest integer
countbygene <- melt(countbygene, id=c("gene")) 
countbygene  <- dcast(countbygene, gene ~ variable, value.var= "value", fun.aggregate=sum)
row.names(countbygene) <- countbygene$gene
countbygene[1] <- NULL
countbygene <- round(countbygene)


# prep for DESeq2
countData <- countbygene 
colData <- samples %>% arrange(RNAseqID)
```

Save Intermediate Data file types, so they could be loaded from here with
StoreReadData
write.table(countData, '../intermediatefiles/countData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(colData, '../intermediatefiles/colData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(geneids, '../intermediatefiles/geneids.csv', row.names = TRUE, sep=",", col.names = T)
read.csv('../intermediatefiles/colData.csv')
read.csv('../intermediatefiles/countData.csv')

```{r SubsetData, echo=FALSE, message=FALSE}
#### now slim the data to get some new figures 
countData <- countbygene 
colData <- samples %>% arrange(RNAseqID)

colData <- colData %>% 
  filter(grepl("Homogenized|Dissociated", Method))  %>% 
  filter(Punch == "DG")  %>% 
  droplevels() ## subsets data
savecols <- as.character(colData$RNAseqID) #select the sample name column that corresponds to row names
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols)) # select just the columns that match the samples in colData
```



#### Differential Gene Expression Plots
```{r DifferentialGeneExpressionAnalysis, echo=FALSE, message=FALSE, warning=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Method )
## filter genes with 0 counts
dds <- dds[ rowSums(counts(dds)) > 1, ]
dds

# Differential expression analysis
dds <- DESeq(dds)
dds

# general deseq
res <- results(dds, independentFiltering = F)
#res
summary(res)
resOrdered <- res[order(res$padj),]
#sum(res$padj < 0.1, na.rm = TRUE)
res05 <- results(dds, alpha=0.05)
summary(res05)
#sum(res05$padj < 0.05, na.rm=TRUE)

## make a plot of fold change as function of expression

plot <- plotMA(res05, main="MA Plot")
return(plot)

resMLE <- results(dds)
#head(resMLE, 4)

## plot most significatnly expressed gene
plotCounts(dds, gene=which.min(res$padj), intgroup="Punch")
# panel2: STXBP6 binds components of the SNARE complex
# panel1: crlf1 Cytokine Receptor Like Factor 1 
plotCounts(dds, gene=which.min(res$padj), intgroup="Method")
# CA1 homodiss only selplg 

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)

## DEG by contrasts
resMethodHomogDiss <- results(dds, contrast = c("Method", "Homogenized", "Dissociated"), independentFiltering = F)
#sum(resMethodHomogDiss$padj < 0.1, na.rm = TRUE) #304 #554 #435
valsMethodHomogDiss <- cbind(resMethodHomogDiss$pvalue, resMethodHomogDiss$padj)
colnames(valsMethodHomogDiss)=c("pval.MethodHomogDiss", "padj.MethodHomogDiss")


#head(assay(rld), 3)
rldd <- assay(rld)
rldpvals <- cbind(rldd, valsMethodHomogDiss)

```
resMethodYokedTrained <- results(dds, contrast = c("Method", "Yoked", "Trained"), independentFiltering = F)
#sum(resMethodYokedTrained$padj < 0.1, na.rm = TRUE) #4
valsMethodYokedTrained <- cbind(resMethodYokedTrained$pvalue, resMethodYokedTrained$padj) 
colnames(valsMethodYokedTrained)=c("pval.MethodYokedTrained", "padj.MethodYokedTrained")

resPunchCA1DG <- results(dds, contrast = c("Punch", "CA1", "DG"), independentFiltering = F)
#sum(resPunchCA1DG$padj < 0.1, na.rm = TRUE) # 4170 #1127
valsPunchCA1DG <- cbind(resPunchCA1DG$pvalue, resPunchCA1DG$padj) 
colnames(valsPunchCA1DG)=c("pval.CA1DG", "padj.CA1DG")

resPunchCA1CA3 <- results(dds, contrast = c("Punch", "CA1", "CA3"), independentFiltering = F)
#sum(resPunchCA1CA3$padj < 0.1, na.rm = TRUE) #2240 # 70
valsPunchCA1CA3 <- cbind(resPunchCA1CA3$pvalue, resPunchCA1CA3$padj) 
colnames(valsPunchCA1CA3)=c("pval.CA1CA3", "padj.CA1CA3")

resPunchCA3DG <- results(dds, contrast = c("Punch", "CA3", "DG"), independentFiltering = F)
#sum(resPunchCA3DG$padj < 0.1, na.rm = TRUE) #4785 #591
valsPunchCA3DG <- cbind(resPunchCA3DG$pvalue, resPunchCA3DG$padj) 
colnames(valsPunchCA3DG)=c("pval.CA3DG", "padj.CA3DG")


#```{r VennDiagram, echo=FALSE, message=FALSE}
rldpvals <- as.data.frame(rldpvals)

MethodHomogDiss <- row.names(rldpvals[rldpvals$padj.MethodHomogDiss<0.1 & !is.na(rldpvals$padj.MethodHomogDiss),])
#MethodYokedTrained <- row.names(rldpvals[rldpvals$padj.valsMethodYokedTrained<0.1 & !is.na(rldpvals$padj.valsMethodYokedTrained),])
PunchCA1DG <- row.names(rldpvals[rldpvals$padj.CA1DG<0.1 & !is.na(rldpvals$padj.CA1DG),])
PunchCA1CA3 <- row.names(rldpvals[rldpvals$padj.CA1CA3<0.1 & !is.na(rldpvals$padj.CA1CA3),])
PunchCA3DG <- row.names(rldpvals[rldpvals$padj.CA3DG<0.1 & !is.na(rldpvals$padj.CA3DG),])

## four way grid
candidates <- list("CA1 v. DG" = PunchCA1DG, "CA1 v. CA3" = PunchCA1CA3, "CA3 v. DG" = PunchCA3DG,  "Homogenized v. Dissociated" = MethodHomogDiss )
dev.off()
prettyvenn <- venn.diagram(
  x = candidates, filename=NULL, lwd=4,
  col = "transparent",
  fill = (values=c("#00441b", "#00441b","#238b45", "#238b45")),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.col = c("darkred", "darkgreen", "blue4", "orange"),
  #cat.dist = c(0.08, 0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

## ca1 ca3 homo diss

candidates <- list("CA1 v. CA3" = PunchCA1CA3,  "Homogenized v. Dissociated" = MethodHomogDiss )
dev.off()
prettyvenn <- venn.diagram(
  x = candidates, filename=NULL, lwd=2,
  col = "transparent",
  fill = (values=c("#00441b", "#00441b")),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.col = c("darkred", "darkgreen", "blue4", "orange"),
  #cat.dist = c(0.08, 0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

## ca1 dg homo diss
candidates <- list("CA1 v. DG" = PunchCA1DG,  "Homogenized v. Dissociated" = MethodHomogDiss )
dev.off()
prettyvenn <- venn.diagram(
  x = candidates, filename=NULL, lwd=4,
  col = "transparent",
  fill = (values=c("#00441b", "#00441b")),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.col = c("darkred", "darkgreen", "blue4", "orange"),
  #cat.dist = c(0.08, 0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

## ca3 dg homo diss
candidates <- list("CA3 v. DG" = PunchCA3DG,  "Homogenized v. Dissociated" = MethodHomogDiss )
dev.off()
prettyvenn <- venn.diagram(
  x = candidates, filename=NULL, lwd=2,
  col = "transparent",
  fill = (values=c("#00441b", "#00441b")),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.col = c("darkred", "darkgreen", "blue4", "orange"),
  #cat.dist = c(0.08, 0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

#```

```{r Heatmap100DEgenes, echo=FALSE, message=FALSE}
nt <- normTransform(dds) # defaults to log2(x+1) 
df <- as.data.frame(colData(dds)[,c("Method", "Punch")])

topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),70)
mat <- assay(rld)[ topVarGenes, ]
mat <- mat - rowMeans(mat)

DEGes <- as.data.frame(rldpvals)
DEGes$rownames <- rownames(DEGes)
DEGes <- DEGes %>%
  filter(padj.MethodHomogDiss < 0.01)
rownames(DEGes) <- DEGes$rownames
drop.cols <- c("padj.MethodHomogDiss", "pval.MethodHomogDiss", "rownames")
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

# pheatmap with 4 groups
ann_colors = list(
  Method=  c(Homogenized = (values=c("#bdbdbd")), Dissociated = (values=c("#636363"))),
  Punch =  c(DG = (values=c("#e7e1ef")),  CA3 = (values=c("#c994c7")), 
             CA1 = (values=c("#dd1c77"))))
# jet color is an attempt to get the matlab color scheme
matlabcolors <-  matlab.like2(100)

pheatmap(DEGes, show_colnames=F, show_rownames = TRUE,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 8, fontsize_row = 6, 
         #cellwidth=10, cellheight=10, width = 10,
         border_color = "grey60" ,
         color = matlabcolors
         )

```

```{r PCA, echo=FALSE, message=FALSE}
source("~/Github/BehavEphyRNAseq/bin/02f_DESeqPCAfunction.R")
plotPCA(rld, intgroup=c("Method", "Punch"), returnData=TRUE)
pcadata <- plotPCA(rld, intgroup=c("Method", "Punch"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

ggplot(pcadata, aes(PC1, PC2, color=Punch, shape=Method)) +
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) 

```