---
title: "DissociationTest.Rmd"
author: "Rayna M Harris"
date: November 20, 2017
#output: html_document
output:
  md_document:
    variant: markdown_github
---

This R Markdown document will walk through the analysis of hippocampal tissue prepared with two different methods. The "homogenized" samples were collected by punch then homogenized in homogenization buffer from the Promega Maxwell kit. The "dissociated samples" were also collected similarily but the cells was dissociated after being punch and before being homogenized. 

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
```


```{r import data, echo=FALSE, message=FALSE}
setwd("~/Github/DissociationTest/bin")
samples <- read.csv("../data/sampleinfo2.csv")
samples$Slice <- as.factor(samples$Slice)
```


```{r wrangle kallisto counts, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
kallistoDirs = dir("../data/JA16444/")
kallistoDirs = kallistoDirs[!grepl("\\.(R|py|pl|sh|xlsx?|txt|tsv|csv|org|md|obo|png|jpg|pdf)$",
        kallistoDirs, ignore.case=TRUE)]
kallistoFiles = paste0(kallistoDirs, "/abundance.tsv")
names(kallistoFiles) = kallistoDirs

setwd('~/Github/DissociationTest/data/JA16444/')
if(file.exists(kallistoFiles))
  kallistoData = lapply(
    kallistoFiles,
    read.table,
    sep = "\t",
    row.names = 1,
    header = TRUE
)


## this for loop uses the reduce function to make two data frame with counts or tpm from all the samples
ids = Reduce(f=union, x=lapply(kallistoData, rownames))
if (all(sapply(kallistoData, function(x) {all(rownames(x)==ids)}))) {
    count = data.frame(
        id = ids,
        sapply(kallistoData, function(x) {x$est_counts}),
        check.names = FALSE,
        stringsAsFactors = FALSE
    )
    tpm = data.frame(
        id = ids,
        sapply(kallistoData, function(x) {x$tpm}),
        check.names = FALSE,
        stringsAsFactors = FALSE
    )
}
 
## make a dataframe with the parts of the gene id as columns
geneids <- count[c(1)] 
geneids$ENSMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 1)
geneids$ENSMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 2)
geneids$OTTMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 3)
geneids$OTTMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 4)
geneids$transcript <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 5)
geneids$gene <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 6)
geneids$length <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 7)
geneids$structure1 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 8)
geneids$structure2 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 9)
geneids$structure3 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 10)
geneids$transcript_lenght <- as.factor(paste(geneids$transcript, geneids$length, sep="_"))

## prep count data for gene level analysis with DESeq2 (as opposed to transcript level anlaysis)
countbygene <- full_join(geneids, count)
countbygene <- countbygene[-c(1:6,8:12)]   ## keep gene name and counts for samples)

## lengthen the dataframe, then wide with gene level sums, then make gene the row name, then round the value to nearest integer
countbygene <- melt(countbygene, id=c("gene")) 
countbygene  <- dcast(countbygene, gene ~ variable, value.var= "value", fun.aggregate=sum)
row.names(countbygene) <- countbygene$gene
countbygene[1] <- NULL
countbygene <- round(countbygene)

# prep for DESeq2
countData <- countbygene 
colData <- samples %>% arrange(RNAseqID)
```

Save Intermediate Data file types, so they could be loaded from here with
StoreReadData
write.table(countData, '../intermediatefiles/countData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(colData, '../intermediatefiles/colData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(geneids, '../intermediatefiles/geneids.csv', row.names = TRUE, sep=",", col.names = T)
read.csv('../intermediatefiles/colData.csv')
read.csv('../intermediatefiles/countData.csv')

write.table(countData, '../intermediatefiles/countData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(colData, '../intermediatefiles/colData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(geneids, '../intermediatefiles/geneids.csv', row.names = TRUE, sep=",", col.names = T)
colData <- read.csv('../intermediatefiles/colData.csv')
countData <- read.csv('../intermediatefiles/countData.csv')


```{r SubsetDatapanel2, echo=FALSE, message=FALSE}
# determines where I want to save a given analysis. Use alternate chuncks for different analyses 
#knitr::opts_chunk$set(fig.path = '../figures/panel2')
#### now slim the data to get some new figures 
#countData <- countbygene 
#colData <- samples %>% arrange(RNAseqID)
## remove outliers
# 146B-DG, 146D-DG, 148B-CA1
#dropcols <- c("146B-DG-2","146D-DG-3", "148B-CA3-4", "146C-CA3-4", "101-DG-3","145B-CA3-1","147D-CA1-1")
#countData <- countData %>% select(-one_of(dropcols))
#colData <- colData %>% filter(!grepl("146B-DG-2|146D-DG-3|148B-CA3|146C-CA3-4|101-DG-3|145B-CA3-1|147D-CA1-1", RNAseqID))

# determines where I want to save a given analysis. Use alternate chuncks for different analyses 
knitr::opts_chunk$set(fig.path = '../figures/panel1')
#### now slim the data to get some new figures 
countData <- countbygene 
colData <- samples %>% arrange(RNAseqID)
## remove outliers
# 146B-DG, 146D-DG, 148B-CA1
countData <- countData[c(1:14)]
colData <- colData %>% filter(grepl("Homogenized|Dissociated", Method))
```



#### Differential Gene Expression Plots
```{r DifferentialGeneExpressionAnalysis, echo=FALSE, message=FALSE, warning=FALSE}
#names(countData)
#names(count)
#rownames(colData)
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Method + Punch + Method*Punch)
## filter genes with 0 counts
dds <- dds[ rowSums(counts(dds)) > 1, ]
#dds

# Differential expression analysis
dds <- DESeq(dds)

# general deseq
res <- results(dds, independentFiltering = F)
#res
resOrdered <- res[order(res$padj),]
#summary(res)
#sum(res$padj < 0.1, na.rm = TRUE)
res05 <- results(dds, alpha=0.05)
#summary(res05)
#sum(res05$padj < 0.05, na.rm=TRUE)

## make a plot of fold change as function of expression
plot <- plotMA(res, main="MA Plot")
return(plot)

resMLE <- results(dds)
#head(resMLE, 4)

## plot most significatnly expressed gene
plotCounts(dds, gene=which.min(res$padj), intgroup="Punch")
# panel2: STXBP6 binds components of the SNARE complex
# panel1: crlf1 Cytokine Receptor Like Factor 1 
plotCounts(dds, gene=which.min(res$padj), intgroup="Method")


## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)

## DEG by contrasts
resMethodHomogDiss <- results(dds, contrast = c("Method", "Homogenized", "Dissociated"), independentFiltering = F)
#sum(resMethodHomogDiss$padj < 0.1, na.rm = TRUE) #304 #256

#resMethodYokedTrained <- results(dds, contrast = c("Method", "Yoked", "Trained"), independentFiltering = F)
#sum(resMethodYokedTrained$padj < 0.1, na.rm = TRUE) #4

resPunchCA1DG <- results(dds, contrast = c("Punch", "CA1", "DG"), independentFiltering = F)
#sum(resPunchCA1DG$padj < 0.1, na.rm = TRUE) # 4170 #778

resPunchCA1CA3 <- results(dds, contrast = c("Punch", "CA1", "CA3"), independentFiltering = F)
#sum(resPunchCA1CA3$padj < 0.1, na.rm = TRUE) #2240 # r

resPunchCA3DG <- results(dds, contrast = c("Punch", "CA3", "DG"), independentFiltering = F)
#sum(resPunchCA3DG$padj < 0.1, na.rm = TRUE) #4785 #556

##  now bind the table of 
valsMethodHomogDiss <- cbind(resMethodHomogDiss$pvalue, resMethodHomogDiss$padj)
#valsMethodYokedTrained <- cbind(resMethodYokedTrained$pvalue, resMethodYokedTrained$padj) 
valsPunchCA1DG <- cbind(resPunchCA1DG$pvalue, resPunchCA1DG$padj) 
valsPunchCA1CA3 <- cbind(resPunchCA1CA3$pvalue, resPunchCA1CA3$padj) 
valsPunchCA3DG <- cbind(resPunchCA3DG$pvalue, resPunchCA3DG$padj) 

colnames(valsMethodHomogDiss)=c("pval.MethodHomogDiss", "padj.MethodHomogDiss")
#colnames(valsMethodYokedTrained)=c("pval.MethodYokedTrained", "padj.MethodYokedTrained")
colnames(valsPunchCA1DG)=c("pval.CA1DG", "padj.CA1DG")
colnames(valsPunchCA1CA3)=c("pval.CA1CA3", "padj.CA1CA3")
colnames(valsPunchCA3DG)=c("pval.CA3DG", "padj.CA3DG")


#head(assay(rld), 3)
rldd <- assay(rld)
rldpvals <- cbind(rldd, valsMethodHomogDiss, valsPunchCA1DG, valsPunchCA1CA3, valsPunchCA3DG)
#head(rldpvals)
#dim(rldpvals)
#17906    46
table(complete.cases(rldpvals))
# 217 17689 
```

```{r VennDiagram, echo=FALSE, message=FALSE}
library(VennDiagram)

rldpvals <- as.data.frame(rldpvals)

MethodHomogDiss <- row.names(rldpvals[rldpvals$padj.MethodHomogDiss<0.1 & !is.na(rldpvals$padj.MethodHomogDiss),])
#MethodYokedTrained <- row.names(rldpvals[rldpvals$padj.valsMethodYokedTrained<0.1 & !is.na(rldpvals$padj.valsMethodYokedTrained),])
PunchCA1DG <- row.names(rldpvals[rldpvals$padj.CA1DG<0.1 & !is.na(rldpvals$padj.CA1DG),])
PunchCA1CA3 <- row.names(rldpvals[rldpvals$padj.CA1CA3<0.1 & !is.na(rldpvals$padj.CA1CA3),])
PunchCA3DG <- row.names(rldpvals[rldpvals$padj.CA3DG<0.1 & !is.na(rldpvals$padj.CA3DG),])

candidates <- list("CA1 v. DG" = PunchCA1DG, "CA1 v. CA3" = PunchCA1CA3, "CA3 v. DG" = PunchCA3DG,  "Homogenized v. Dissociated" = MethodHomogDiss )

prettyvenn <- venn.diagram(
  x = candidates, filename=NULL, lwd=4,
  col = "transparent",
  fill = (values=c("#00441b", "#00441b","#238b45", "#238b45")),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.col = c("darkred", "darkgreen", "blue4", "orange"),
  #cat.dist = c(0.08, 0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans"
);
grid.draw(prettyvenn)

```

```{r Heatmap100DEgenes, echo=FALSE, message=FALSE}
library("pheatmap")
nt <- normTransform(dds) # defaults to log2(x+1) 
df <- as.data.frame(colData(dds)[,c("Method","Punch")])
#str(df)

ann_colors = list(
  Method=  c(Homogenized = (values=c("#e34a33")), Dissociated = (values=c("#fdbb84")),
  Punch =  c(DG = (values=c("#006837")),  CA3 = (values=c("#41ab5d")), 
             CA1 = (values=c("#d9f0a3")))))


library("genefilter")
topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),50)
#topVarGenes <- Method
mat <- assay(rld)[ topVarGenes, ]
mat <- mat - rowMeans(mat)


pheatmap(mat, show_colnames=TRUE, show_rownames = TRUE,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 12, fontsize_row = 10, 
         #cellwidth=10, cellheight=10,
         #width = 10,
         border_color = "grey60"
         )

```

```{r PCA, echo=FALSE, message=FALSE}
source("~/Github/BehavEphyRNAseq/bin/02f_DESeqPCAfunction.R")
plotPCA(rld, intgroup=c("Punch.Collector", "Punch"), returnData=TRUE)
pcadata <- plotPCA(rld, intgroup=c("Punch.Collector", "Punch"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

ggplot(pcadata, aes(PC1, PC2, color=Punch, shape=Punch.Collector)) +
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))


plotPCA(rld, intgroup=c("Method", "Punch"), returnData=TRUE)
pcadata <- plotPCA(rld, intgroup=c("Method", "Punch"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

ggplot(pcadata, aes(PC1, PC2, color=Punch, shape=Method)) +
  geom_point(size=5) + 
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))

```