---
title: "DissociationTest.Rmd"
author: "Rayna M Harris"
date: November 20, 2017
#output: html_document
output:
  md_document:
    variant: markdown_github
---
## Methods for Dorsal Hippocampal Gene Expression Profiling

#### Part 1: Examining the influence of dissasociation on gene expression in the CA1, CA3, and DG 


```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(colorRamps) # for a matlab like color scheme
library(cowplot)

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/allregions_onlyhomodiss/')
```

```{r ImportData, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
# this starts with data genearated from code described in KallistoGather.Rmd
colData <- read.csv('../intermediatefiles/colData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../intermediatefiles/countData.csv', check.names = F)

```

```{r SubsetData, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#### now slim the data to get some new figures 
colData <- colData %>% 
  filter(grepl("100|101", Mouse))  %>% 
  #filter(RNAseqID != "101-DG-3")  %>% 
  droplevels() ## subsets data
savecols <- as.character(colData$RNAseqID) #select the sample name column that corresponds to row names
savecols <- as.vector(savecols) # make it a vector

countData <- countData %>% select(one_of(savecols)) # select just the columns that match the samples in colData
## remove counts with count value  < 2
countData[countData < 2] <- 0

```


```{r DifferentialGeneExpressionAnalysis, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Method + Punch + Method * Punch )
## filter genes with 0 counts
dds <- dds[ rowSums(counts(dds)) > 2, ]
dds

# Differential expression analysis
dds <- DESeq(dds)
dds

# general deseq
res <- results(dds, independentFiltering = F)
#res
summary(res)
resOrdered <- res[order(res$padj),]
sum(res$padj < 0.1, na.rm = TRUE)
res05 <- results(dds, alpha=0.05)
summary(res05)
sum(res05$padj < 0.05, na.rm=TRUE)

## make a plot of fold change as function of expression
#plot <- plotMA(res05, main="MA Plot")
#return(plot)

resMLE <- results(dds)
#head(resMLE, 4)

## plot most significatnly expressed gene
#plotCounts(dds, gene=which.min(res$padj), intgroup="Punch")
#plotCounts(dds, gene=which.min(res$padj), intgroup="Method")

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)

## DEG by contrasts
## this calls the function I wrote in resvalsfunction.R
source("resvalsfunction.R")
valsMethodHomogDiss <- resvals(contrastvector = c('Method', 'Homogenized', 'Dissociated'), mypadj = 0.01)
valsPunchCA1DG <- resvals(contrastvector = c('Punch', 'CA1', 'DG'), mypadj = 0.01)
valsPunchCA1CA3 <- resvals(contrastvector = c('Punch', 'CA1', 'CA3'), mypadj = 0.01)
valsPunchCA3DG <- resvals(contrastvector = c('Punch', 'CA3', 'DG'), mypadj = 0.01)

#head(assay(rld), 3)
rldd <- assay(rld)
rldpvals <- cbind(rldd, valsMethodHomogDiss, valsPunchCA1DG, valsPunchCA1CA3, valsPunchCA3DG)

```

This PCA gives an overview of the variability between samples using the a large matrix of log transformed gene expression data. You can see that the bigges difference is between DG punches and the CA1 and CA3 punches. CA1 and CA3 samples have similar transcriptomes. The homogenized CA1 samples have the most similar transcriptonal profiles as evidenced by their tight clustering. 

```{r PCA, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
source("DESeqPCAfunction.R")
plotPCA(rld, intgroup=c("Method", "Punch"), returnData=TRUE)
pcadata <- pcaplot16(rld, intgroup=c("Method", "Punch"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

pcamethod <- ggplot(pcadata, aes(PC1, PC2, color=Method, label=name)) +
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  theme_classic() +
  stat_ellipse(level = 0.95, (aes(color=Method)),size=1)  

pcamethod <- ggplot(pcadata, aes(PC1, PC3, color=Method, label=name)) +
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC3: ",percentVar[3],"% variance")) +
  theme_classic() +
  stat_ellipse(level = 0.95, (aes(color=Method)),size=1)  

pcamethod <- ggplot(pcadata, aes(PC2, PC3, color=Method, label=name)) +
  geom_point(size=5) +
  xlab(paste0("PC2: ",percentVar[2],"% variance")) +
  ylab(paste0("PC3: ",percentVar[3],"% variance")) +
  theme_classic() +
  stat_ellipse(level = 0.95, (aes(color=Method)),size=1) 


pcaregion <- ggplot(pcadata, aes(PC1, PC2, color=Punch, label=name)) +
  geom_point(size=5) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  theme_classic() +
  stat_ellipse(level = 0.95, (aes(color=Punch)),size=1) +
  scale_color_manual(values=c("#08306b", "#238443", "#a50f15"))

plot2by2 <- plot_grid(pcamethod,pcaregion,
                      labels=c("A", "B"), ncol = 2)
pcamethod
pcaregion
plot2by2
save_plot("../figures/allregions_onlyhomodiss/plot2by2.pdf", plot2by2,
          ncol = 2, # we're saving a grid plot of 2 columns
          nrow = 1)

save_plot("../figures/allregions_onlyhomodiss/pcamethod.png", pcamethod)
save_plot("../figures/allregions_onlyhomodiss/pcaregion.png", pcaregion)

```


This Venn Diagram shows the number of differentially expressed by contrast described above each oval. The most number of genes are differntially expressed between DG and the CAs (nearly 1000) wheras only about 200 were differntailly regulated as a result of of technical maniplulation comparing homogenized and dissociated samples. 

```{r VennDiagram, echo=FALSE, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
rldpvals <- as.data.frame(rldpvals)

MethodHomogDiss <- row.names(rldpvals[rldpvals$padjMethodHomogenizedDissociated<0.1 & !is.na(rldpvals$padjMethodHomogenizedDissociated),])
PunchCA1DG <- row.names(rldpvals[rldpvals$padjPunchCA1DG <0.1 & !is.na(rldpvals$padjPunchCA1DG),])
PunchCA1CA3 <- row.names(rldpvals[rldpvals$padjPunchCA1CA3 <0.1 & !is.na(rldpvals$padjPunchCA1CA3),])
PunchCA3DG <- row.names(rldpvals[rldpvals$padjPunchCA3DG <0.1 & !is.na(rldpvals$padjPunchCA3DG),])

## four way grid
candidates <- list("CA1 v. DG" = PunchCA1DG, "CA1 v. CA3" = PunchCA1CA3, "Homogenized v. Dissociated" = MethodHomogDiss )
#dev.off()
prettyvenn <- venn.diagram(lwd =3,
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.dist = c(0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)


plot.new()

candidates <- list("CA1 v. DG" = PunchCA1DG, "CA1 v. CA3" = PunchCA1CA3, "CA3 v. DG" = PunchCA3DG,  "Homogenized v. Dissociated" = MethodHomogDiss )
#dev.off()
prettyvenn <- venn.diagram(lwd =3,
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.dist = c(0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

save_plot("../figures/allregions_onlyhomodiss/prettyvenn.pdf", prettyvenn,
          ncol = 1, # we're saving a grid plot of 2 columns
          nrow = 1)



```

I'm not really happy with these two heat maps. Here's how I created them. 
Top heatmap:  subset the data to give only the gene with an adjusted p value < 0.05 for the homogenized vs dissociated comparisonany two-way comparsion. 
Bottom heatmap: subset the data to give only the gene with an adjusted p value < 0.05 for two way brain region comparision (CA1 vs DG, CA3, vs DG, or CA1 vs DG)

Here, you can see that the differences between samples is not as clear cut for all comparisions. What other mechanisms would be useful for subseting the data to identify genes of interest?

```{r Heatmap100DEgenes, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
nt <- normTransform(dds) # defaults to log2(x+1) 
df <- as.data.frame(colData(dds)[,c("Method", "Punch")])
ann_colors = list(Method = c(Homogenized = (values=c("#969696")), Dissociated = (values=c("#525252"))),
  Punch =  c(DG = (values=c("#a50f15")),  CA3 = (values=c("#238443")),
             CA1 = (values=c("#08306b"))))
matlabcolors <-  matlab.like2(100)  #color schelem
cembrowskicolors <-  colorRampPalette(c("Deep Sky Blue 3", "white", "red"))( 30 )

#topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),70)
#mat <- assay(rld)[ topVarGenes, ]
#mat <- mat - rowMeans(mat)

DEGes <- as.data.frame(rldpvals) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$padjmin <- with(DEGes, pmin(padj.MethodHomogDiss)) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.01)
rownames(DEGes) <- DEGes$rownames
drop.cols <- c("padj.MethodHomogDiss", "pval.MethodHomogDiss", "padj.CA1DG" , "padj.CA1CA3" , "padj.CA3DG" , "pval.CA1DG" , "pval.CA1CA3" , "pval.CA3DG" , "rownames", "padjmin")
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
#DEGes <- DEGes - rowMeans(DEGes)

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 12, fontsize_row = 7, 
         #cellwidth=10, cellheight=10, width = 10,
         border_color = "grey60" ,
         color = cembrowskicolors,
         main = "padj.MethodHomogDiss < 0.01",
         filename = "../figures/allregions_onlyhomodiss/myheatmap.pdf")

DEGes <- as.data.frame(rldpvals) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$padjmin <- with(DEGes, pmin(padj.CA1DG, padj.CA1CA3, padj.CA3DG)) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.005)
rownames(DEGes) <- DEGes$rownames
drop.cols <- c("padj.MethodHomogDiss", "pval.MethodHomogDiss", "padj.CA1DG" , "padj.CA1CA3" , "padj.CA3DG" , "pval.CA1DG" , "pval.CA1CA3" , "pval.CA3DG" , "rownames", "padjmin")
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
#DEGes <- DEGes - rowMeans(DEGes)

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 12, fontsize_row = 6, 
         #cellwidth=10, cellheight=10, width = 10,
         border_color = "grey60" ,
         color = cembrowskicolors,
         main = "Hippocampal Region padj < 0.005"
         )
```



This is a data validation check plot. Here, I'm showing how many millions of reads were present in each sample. On average, each sample had 5 million reads, but the range was from 0.8 to 10 millino reads. 
```{r edgeR, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
library(edgeR)

counts <- countData
dim( counts )
colSums( counts ) / 1e06  # in millions of gene counts
table( rowSums( counts ) )[ 1:30 ] # Number of genes with low counts
```

```{r joinwithgeneids, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
#geneids <- read.csv('../intermediatefiles/geneids.csv')
#rownames(colData) <- colData$RNAseqID
#listofDegs <- row.names(DEGes)
#listofDegs <- as.data.frame(listofDegs)
#names(listofDegs)[names(listofDegs)=="listofDegs"] <- "gene"
#listofDegs <- left_join(listofDegs, geneids) 
#listofDegs <- listofDegs %>% 
 # distinct(gene, ENSMUSG) 
#rownames(listofDegs) <- listofDegs$gene 
#listofDegs <- listofDegs %>% select(ENSMUSG) 

#write.table(listofDegs, '../intermediatefiles/listofDegs.csv', row.names = TRUE, sep=",", col.names = T)
```


```{r readcounts, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
readcounts <- read.table("../intermediatefiles/slimreadcounts.txt")
str(readcounts)
library(dplyr)
library(ggplot2)
mean(readcounts$V2) #4918879
sum(readcounts$V2)  #68864306

readcounts$millionreads <- readcounts$V2/1000000

ggplot(readcounts, aes(x=millionreads)) + 
  geom_histogram(binwidth = 1, colour = "black", fill = "darkgrey") +
  theme_classic() +
  scale_x_continuous(name = "Millions of Reads per Sample",
                           breaks = seq(0, 10, 1),
                           limits=c(0, 10)) +
  scale_y_continuous(name = "Number of Samples")

```

This graph examines the magnitude of gene expression differences (shown as log fold change on the y axis) as a function of read abundance (shown as mean normalized counts on the x axis.  
```{r MAplot, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
## make a plot of fold change as function of expression
plot <- plotMA(res05, main="MA Plot")
return(plot)
```

This is the gene with the most significant p value when you compare DG dissociated to the rest of the samples. 
```{r mostsignificantgene, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
## plot most significatnly expressed gene
plotCounts(dds, gene=which.min(res$padj), intgroup="Punch")
plotCounts(dds, gene=which.min(res$padj), intgroup="Method")

```


```{r scatterplots, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
par( mfrow = c( 1, 3 ) )
plot( assay(rld)[, 1:2], col="#00000020", pch=20, cex=0.3 )
plot( assay(rld)[, 4:5], col="#00000020", pch=20, cex=0.3 )
plot( assay(rld)[, 6:7], col="#00000020", pch=20, cex=0.3 )

sampleDists <- dist( t( assay(rld) ) )
sampleDists

sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( rld$treatment,
rld$patient, sep="-" )
colnames(sampleDistMatrix) <- NULL
library( "gplots" )
library( "RColorBrewer" )
colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours)

library(ggcorrplot)
mycormat <- cor(assay(rld))
mycormat
ggcorrplot(mycormat)
ggcorrplot(mycormat, hc.order = TRUE, type = "lower",
   lab = TRUE)
```