---
title: "DissociationTest.Rmd"
author: "Rayna M Harris"
date: November 20, 2017
#output: html_document
output:
  md_document:
    variant: markdown_github
---

This R Markdown document will walk through the analysis of hippocampal tissue prepared with two different methods. The "homogenized" samples were collected by punch then homogenized in homogenization buffer from the Promega Maxwell kit. The "dissociated samples" were also collected similarily but the cells was dissociated after being punch and before being homogenized. 

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(magrittr)
library(tidyverse)
library(reshape2)
```


```{r import data, echo=FALSE, message=FALSE}
setwd("~/Github/DissociationTest/bin")
samples <- read.csv("../data/sampleinfo2.csv")
samples$Slice <- as.factor(samples$Slice)
```


```{r wrangle kallisto counts, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
kallistoDirs = dir("../data/JA16444/")
kallistoDirs = kallistoDirs[!grepl("\\.(R|py|pl|sh|xlsx?|txt|tsv|csv|org|md|obo|png|jpg|pdf)$",
        kallistoDirs, ignore.case=TRUE)]
kallistoFiles = paste0(kallistoDirs, "/abundance.tsv")
names(kallistoFiles) = kallistoDirs

setwd('~/Github/DissociationTest/data/JA16444/')
if(file.exists(kallistoFiles))
  kallistoData = lapply(
    kallistoFiles,
    read.table,
    sep = "\t",
    row.names = 1,
    header = TRUE
)


## this for loop uses the reduce function to make two data frame with counts or tpm from all the samples
ids = Reduce(f=union, x=lapply(kallistoData, rownames))
if (all(sapply(kallistoData, function(x) {all(rownames(x)==ids)}))) {
    count = data.frame(
        id = ids,
        sapply(kallistoData, function(x) {x$est_counts}),
        check.names = FALSE,
        stringsAsFactors = FALSE
    )
    tpm = data.frame(
        id = ids,
        sapply(kallistoData, function(x) {x$tpm}),
        check.names = FALSE,
        stringsAsFactors = FALSE
    )
}
 
## make a dataframe with the parts of the gene id as columns
geneids <- count[c(1)] 
geneids$ENSMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 1)
geneids$ENSMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 2)
geneids$OTTMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 3)
geneids$OTTMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 4)
geneids$transcript <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 5)
geneids$gene <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 6)
geneids$length <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 7)
geneids$structure1 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 8)
geneids$structure2 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 9)
geneids$structure3 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 10)
geneids$transcript_lenght <- as.factor(paste(geneids$transcript, geneids$length, sep="_"))

## prep count data for gene level analysis with DESeq2 (as opposed to transcript level anlaysis)
countbygene <- full_join(geneids, count)
countbygene <- countbygene[-c(1:6,8:12)]   ## keep gene name and counts for samples)
## lengthen the dataframe, then wide with gene level sums, then make gene the row name, then round the value to nearest integer
countbygene <- melt(countbygene, id=c("gene")) 
countbygene  <- dcast(countbygene, gene ~ variable, value.var= "value", fun.aggregate=sum)
row.names(countbygene) <- countbygene$gene
countbygene[1] <- NULL
countbygene <- round(countbygene)
# prep for DESeq2
countData <- countbygene 
colData <- samples %>% arrange(RNAseqID)


## prep count data for gene level analysis with DESeq2 (as opposed to transcript level anlaysis)
tpmbygene <- full_join(geneids, tpm)
tpmbygene <- tpmbygene[-c(1:6,8:12)]   ## keep gene name and counts for samples)
## lengthen the dataframe, then wide with gene level sums, then make gene the row name, then round the value to nearest integer
tpmbygene <- melt(tpmbygene, id=c("gene")) 
tpmbygene  <- dcast(tpmbygene, gene ~ variable, value.var= "value", fun.aggregate=sum)
row.names(tpmbygene) <- tpmbygene$gene
tpmbygene[1] <- NULL
tpmbygene <- round(tpmbygene)
# prep for DESeq2
tpmData <- tpmbygene 


```

Save Intermediate Data file types, so they could be loaded from here with
StoreReadData
write.table(countData, '../intermediatefiles/countData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(colData, '../intermediatefiles/colData.csv', row.names = TRUE, sep=",", col.names = T)
write.table(geneids, '../intermediatefiles/geneids.csv', row.names = TRUE, sep=",", col.names = T)
write.table(tpmData, '../intermediatefiles/tpmData.csv', row.names = TRUE, sep=",", col.names = T)


