---
title: "01_DissociationTest.Rmd"
author: "Rayna M Harris"
date: March 11, 2017
output: md_document
---
## Cellular Stress

In this analysis, I examine the effect that cell dissasociation has on CA1, CA3, and DG gene expression relative to homogenized tissue samples.

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(cowplot)
library(RColorBrewer)
library(ggcorrplot)
library(dplyr)
library(ggplot2)
library(colorRamps)

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/01_dissociationtest/')
```

```{r ImportData, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
# this starts with data genearated from code described in ../../BehavEphysRNAseq/markdownfiles/02a_tidysamples.R and ../../BehavEphysRNAseq/markdownfiles/02b_KallistoGather.Rmd
colData <- read.csv('../data/DissociationColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/DissociationCountData.csv', check.names = F, row.names = 1)
```


```{r SubsetData, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
## Subset to just look homogenized and dissociated samples
colData <- colData %>%
  filter(Mouse %in% c("15-100")) %>% droplevels()
savecols <- as.character(colData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols)) # keep good samples
```

Here is a brief overview of the samples being compared.

```{r sampleinfo}
str(colData)
summary(colData)
```


```{r DEG, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Method + Region + Method * Region )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## filter genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis
dds

# general deseq
res <- results(dds, independentFiltering = F)

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- getVarianceStabilizedData(dds) 
```


This PCA gives an overview of the variability between samples using the a large matrix of log transformed gene expression data. You can see that the bigges difference is between DG punches and the CA1 and CA3 punches. CA1 and CA3 samples have similar transcriptomes. The homogenized CA1 samples have the most similar transcriptonal profiles as evidenced by their tight clustering. 

```{r PCA, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
source("DESeqPCAfunction.R")
source("figureoptions.R")

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Region", "Method"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

## for markdown
plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", colorvalues = colorvalRegion, aesshape = pcadata$Method, shapename = "Method")

# for adobe
myplot <- plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Method, shapename = "Method", colorvalues = colorvalRegion)
pdf(file="../figures/01_dissociationtest/PCA-1.pdf", width=4.5, height=3)
plot(myplot)
dev.off()
```

Now, we can calulate the number of significant genes by contrast by contrast. The first number displayed is not corrected for mutiple hypothesis testing but the second one is.

```{r signficiantgenes}
## DEG by contrasts
source("resvalsfunction.R")
contrast1 <- resvals(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.1)
contrast2 <- resvals(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.1)
contrast3 <- resvals(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.1)
contrast4 <- resvals(contrastvector = c('Method', 'control', 'dissociated'), mypval = 0.1)
```


Now, we can view a histogram of the distribution 

```{r histogram, echo=FALSE}
source("resvalsfunction.R")
myhistogram(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.1)
myhistogram(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.1)
myhistogram(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.1)
myhistogram(contrastvector = c('Method', 'control', 'dissociated'), mypval = 0.1)

```

This Venn Diagram sthe overlap of differentailly expression genes by Region and method. This shows all genes with *uncorrected* pvalue <0.1.

```{r VennDiagramPVal, echo=FALSE, echo=FALSE, message=TRUE, comment=FALSE, warning=FALSE, results='hide'}

#create a new DF with the gene counts
rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast1, contrast2, contrast3, contrast4)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]

# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[1] <0.1 & !is.na(rldpvals[1]),])
venn2 <- row.names(rldpvals[rldpvals[3] <0.1 & !is.na(rldpvals[3]),])
venn3 <- row.names(rldpvals[rldpvals[5] <0.1 & !is.na(rldpvals[5]),])
venn4 <- row.names(rldpvals[rldpvals[7] <0.1 & !is.na(rldpvals[7]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)


## check order for correctness
candidates <- list("Region" = venn123, "Method" = venn4)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)

```

This Venn Diagram sthe overlap of differentailly expression genes by Region and method. This shows all genes with *adjusted* pvalue <0.1.

```{r VennDiagramPadj, echo=FALSE}
#create a new DF with the gene counts
rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast1, contrast2, contrast3, contrast4)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]


# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[2] <0.1 & !is.na(rldpvals[2]),])
venn2 <- row.names(rldpvals[rldpvals[4] <0.1 & !is.na(rldpvals[4]),])
venn3 <- row.names(rldpvals[rldpvals[6] <0.1 & !is.na(rldpvals[6]),])
venn4 <- row.names(rldpvals[rldpvals[8] <0.1 & !is.na(rldpvals[8]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)

# save files for big venn diagram
write(venn123, "../results/01_dissociation_venn123.txt")
write(venn4, "../results/01_dissociation_venn4.txt")



## check order for correctness
candidates <- list("Region" = venn123, "Method" = venn4)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)

```

#heatmaps

```{r HeatmapPadj, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
## Any padj <0.1
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

DEGes$padjmin <- with(DEGes, pmin(padjMethodcontroldissociated, padjRegionCA1DG ,padjRegionCA3DG, padjRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.1)

#DEGes$pvalmin <- with(DEGes, pmin(pvalMethodcontroldissociated, pvalRegionCA1DG ,pvalRegionCA3DG, pvalRegionCA1CA3 )) # put the min pvalue in a new column
#DEGes <- DEGes %>% filter(pvalmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)


# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsdissociation
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Method", "Region")])
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))


pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 12, fontsize_row = 7, 
         #cellwidth=10, cellheight=10, width = 10,
         border_color = "grey60" ,
         color = colorpalette,
         #main = "Any Padj < 0.05",
         breaks=myBreaks
         )

```

```{r HeatmapPvalue, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
## Any pvalue <0.1
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

#DEGes$padjmin <- with(DEGes, pmin(padjMethodcontroldissociated, padjRegionCA1DG ,padjRegionCA3DG, padjRegionCA1CA3 )) # put the min pvalue in a new column
#DEGes <- DEGes %>% filter(padjmin < 0.1)

DEGes$pvalmin <- with(DEGes, pmin(pvalMethodcontroldissociated, pvalRegionCA1DG ,pvalRegionCA3DG, pvalRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(pvalmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)


# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsdissociation
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Method", "Region")])

paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

# for markdown file
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         #width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         #cellwidth = 12, 
         #main = "Any Pval < 0.1",
         breaks=myBreaks
         )

# for adobe
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         cellwidth = 12, 
         #main = "Any Pval < 0.1",
         filename = "../figures/01_dissociationtest/HeatmapPvalue-1.pdf",
         breaks=myBreaks
         )
```


This shows how strongly correlated the samples are. Even across brain regions, correlation is super fucking hight.

```{r sampleheatmap, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE }
rld <- rlogTransformation(dds)
#head(assay(rld))
colnames(rld) <- paste(colData$Region, colData$Method, colData$RNAseqID, sep = "")
#head(assay(rld))
pheatmap(cor(assay(rld)),border_color=NA, main="Sample Heatmap Log Transformed Data", show_colnames=F, show_rownames = F)

vsd <- getVarianceStabilizedData(dds) 
pheatmap(cor(vsd),border_color=NA, main="Sample Heatmap Variance Stabilized Data", show_colnames=F, show_rownames = F)
```


This is a data validation check plot. Here, I'm showing how many millions of reads were present in each sample. On average, each sample had 5 million reads, but the range was from 0.8 to 10 millino reads. 

```{r edgeR, echo=FALSE, message=FALSE,comment=FALSE, warning=FALSE }
library(edgeR)
counts <- countData
dim( counts )
colSums( counts ) / 1e06  # in millions of gene counts
table( rowSums( counts ) )[ 1:30 ] # Number of genes with low counts
```

Save files for GO analysis.
A total of 217 DEGs with unadjusted p-value < 0.1 were input into the GO anlaysis.

```{r GOsetup, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE }
# from https://github.com/rachelwright8/Ahya-White-Syndromes/blob/master/deseq2_Ahya.R

resCD=results(dds, contrast=c('Method', 'control', 'dissociated'))
table(resCD$padj<0.1)
table(resCD$pvalue<0.05)
head(resCD)
logs <- data.frame(cbind("gene"=row.names(resCD),"logP"=round(-log(resCD$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[resCD$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign

write.table(logs,quote=F,row.names=F,file="./06_GO_MWU/01_dissociation_GOpvals.csv",sep=",")

```

