---
title: "01_DissociationTest.Rmd"
author: "Rayna M Harris"
date: March 11, 2017
output: md_document
---
### Identifying the effects of cellular dissociation on hippocampal transcriptomes

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(cowplot)
library(RColorBrewer)
library(dplyr)
library(plyr)
library(ggplot2)
library(colorRamps)
library(car)

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/01_dissociationtest/')
```

```{r ImportData}
# this starts with data genearated from code described in ../../BehavEphysRNAseq/markdownfiles/02a_tidysamples.R and ../../BehavEphysRNAseq/markdownfiles/02b_KallistoGather.Rmd and  ../../BehavEphysRNAseq/markdownfiles/02d_subsetforprojects.Rmd
colData <- read.csv('../data/DissociationColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/DissociationCountData.csv', check.names = F, row.names = 1)
```

```{r SubsetData}
## Subset to just look control and dissociated samples
colData <- colData %>%
  filter(Mouse %in% c("15-100")) %>% droplevels()
savecols <- as.character(colData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols)) # keep good samples

## rename and relevel things
colData <- rename(colData, c("Method"="Treatment"))
```

Sample sizes

```{r viewcoldata}
colData %>% select(Treatment,Region)  %>%  summary()
```

```{r viewscountData}
dim(countData)
```

This is a supplementary data validation check plot. Here, I'm showing how many millions of reads were present in each sample. On average, each sample had 5 million reads, but the range was from 0.8 to 10 millino reads. 

```{r edgeR}
library(edgeR)
counts <- countData
dim( counts )
colSums( counts ) / 1e06  # in millions of gene counts
table( rowSums( counts ) )[ 1:30 ] # Number of genes with low counts
```

I used DESeq2 (Love et al., 2014) for gene expression normalization and quantification using the following experimental design: `Treatment + Region + Treatment * Region`. Genes with less than 2 counts across all samples were filtered, leaving us with `dim(rld)` number of genes for analysis of differntial expression.

```{r DEG}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Treatment + Region + Treatment * Region )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## filter genes 
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data
dim(rld) #print total genes analyzed
```

We identified 162 genes that were differentially expressed between the control and dissociated samples, 331 genes that were differentially expressed genes (DEGs) between any of the three hippocampus subfields, and 30 genes were shared between both sets of differentially expressed genes at FDR p-value < 0.05 (Fig 1B). 


```{r signficiantgenes}
## DEG by contrasts
source("resvalsfunction.R")
contrast1 <- resvals(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.05)
contrast2 <- resvals(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.05)
contrast3 <- resvals(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.05)
contrast4 <- resvals(contrastvector = c('Treatment', 'dissociated', 'control'), mypval = 0.05)
```


```{r VennDiagramPadj}
#create a new DF with the gene counts
rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast1, contrast2, contrast3, contrast4)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]


# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[2] <0.05 & !is.na(rldpvals[2]),])
venn2 <- row.names(rldpvals[rldpvals[4] <0.05 & !is.na(rldpvals[4]),])
venn3 <- row.names(rldpvals[rldpvals[6] <0.05 & !is.na(rldpvals[6]),])
venn4 <- row.names(rldpvals[rldpvals[8] <0.05 & !is.na(rldpvals[8]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)

## check order for correctness
candidates <- list("Region" = venn123, "Treatment" = venn4)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)

# save files for metanalysis
write(venn123, "../results/01_dissociation_venn123.txt")
write(venn4, "../results/01_dissociation_venn4.txt")
write(venn1, "../results/01_dissociation_venn1.txt")

```


A hierarchical clustering analysis of all differentially expressed genes does not give rise to distinct clusters that are separated by subfield or method; however, when examining the control, homogenized samples alone (identified with light grey boxes), the three subfields form distinct clusters, while the dissociated samples do not cluster by subfield (Fig. 1C). 


```{r HeatmapPadj}
## Any padj <0.05
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

DEGes$padjmin <- with(DEGes, pmin(padjTreatmentdissociatedcontrol, padjRegionCA1DG ,padjRegionCA3DG, padjRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.05)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsdissociation
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Treatment", "Region")])
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))


pheatmap(DEGes, show_colnames=T, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 8, fontsize_row = 7, 
         cellwidth=20,
         border_color = "grey60" ,
         color = colorpalette,
         breaks=myBreaks,
         clustering_method="average",
         clustering_distance_cols="correlation"
         )

# for adobe
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 8, 
         width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         cellwidth = 8, 
         breaks=myBreaks,
         clustering_method="average",
         clustering_distance_cols="correlation" ,
         filename = "../figures/01_dissociationtest/HeatmapPadj-1.pdf"
         )
```


```{r pvclust, include=FALSE}
library(pvclust)
# clustering just the degs
result <- pvclust(DEGes, method.dist="cor", method.hclust="average", nboot=1000)
plot(result)
```


This PCA gives an overview of the variability between samples using the a large matrix of log transformed gene expression data. You can see that the biggest difference is between DG punches and the CA1 and CA3 punches. CA1 and CA3 samples have similar transcriptomes. The control CA1 samples have the most similar transcriptonal profiles as evidenced by their tight clustering. 

```{r PCA, message=FALSE, warning=FALSE}
source("DESeqPCAfunction.R")
source("figureoptions.R")

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Region", "Treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

## for markdown
plotPC2PC1(aescolor = pcadata$Region, colorname = "Region", colorvalues = colorvalRegion, aesshape = pcadata$Treatment, shapename = "Treatment")

# for adobe
myplot <- plotPC2PC1(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)
pdf(file="../figures/01_dissociationtest/PCA-1.pdf", width=4.5, height=3)
plot(myplot)
dev.off()

## statistics
aov1 <- aov(PC1 ~ Region, data=pcadata)
summary(aov1) 
TukeyHSD(aov1, which = "Region") 

aov2 <- aov(PC2 ~ Region, data=pcadata)
summary(aov2) 
TukeyHSD(aov2, which = "Region") 

aov3 <- aov(PC1 ~ Treatment, data=pcadata)
summary(aov3) 
TukeyHSD(aov3, which = "Treatment")

aov4 <- aov(PC2 ~ Treatment, data=pcadata)
summary(aov4) 
TukeyHSD(aov4, which = "Treatment") 

```


Next, save files for dowstream GO analysis.

```{r GOsetup}
# from https://github.com/rachelwright8/Ahya-White-Syndromes/blob/master/deseq2_Ahya.R

resCD=results(dds, contrast=c('Treatment', 'dissociated', 'control'), independentFiltering = F)
table(resCD$padj<0.05)

logs <- data.frame(cbind("gene"=row.names(resCD),"logP"=round(-log(resCD$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[resCD$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign

write.csv(logs, file = "./06_GO_MWU/01_dissociation_GOpvals.csv", row.names = F)
```



Supplementary histograms  
```{r histogram, include=FALSE}
source("resvalsfunction.R")
myhistogram(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.05)
myhistogram(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.05)
myhistogram(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.05)
myhistogram(contrastvector = c('Treatment', 'dissociated', 'control'), mypval = 0.05)

```