---
title: "03_cognitiontest.Rmd"
author: "Rayna M Harris"
date: "March 6, 2017"
output: md_document
---

```{r setup, include=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(cowplot)
library(RColorBrewer)
library(dplyr)
library(plyr)
library(ggplot2)
library(edgeR)
library(knitr) 
library(pvclust)
library(knitr) 

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/03_cognitiontest/')

# user defined funcitons and options
source("resvalsfunction.R") 
source("DESeqPCAfunction.R")
source("figureoptions.R")
```

## Examining of cognitive training on hippocampal
The goals of the subsequent analysis are 1) to determine the effects of cognitiving training on hippocampal gene expression and 2) related any detectable changes to variation cause by other technical and biological treatements.

The sample and count information for this part is found in `../data/StressCognitionTestColData.csv` and `../data/StressCognitionTestCountData.csv`. You can also download these two files (with a different name but same content) from [GEO GSE100225 ](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE100225). 

```{r importdata, include=FALSE}
colData <- read.csv('../data/StressCognitionTestColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/StressCognitionTestCountData.csv', check.names = F)

colData <- colData %>%
  filter(Treatment %in% c("yoked", "trained")) %>% droplevels()

savecols <- as.character(colData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols)) # keep good samples

## rename and relevel things
colData$Treatment <- factor(colData$Treatment, levels = c("yoked", "trained"))
```

### Experimental Design
We use 3-4â€“month-old male C57BL/6J mice fro the Jackson Laboratory and housed at the Marine Biological Laboratory. Mice (N=4) trained in the active place avoidance task are conditioned to avoid mild shocks that can be localized by visual cues in the enviornment. Yoked control mice (N=4) are delivered sequence of unavoidable shock that mimickes the time series of shocks received by the trained mice. While the trained and yoked animals received the same number of shocks, only the trained animals exhibitied an avoidance response. 

Thirty minutes after the last cognitive training session, mice were killed and transverse brain slices were prepared. The DG, CA3, CA1 subregions were microdissected using a 0.25 mm punch (Electron Microscopy Systems) and a dissecting scope (Zeiss). RNA was isolated using the Maxwell 16 LEV RNA Isolation Kit (Promega). RNA libraries were prepared by the Genomic Sequencing and Analysis Facility at the University of Texas at Austin using the Illumina HiSeq platform. 

```{r biologicalsampleimage, echo=FALSE, warning=FALSE}
include_graphics('../figures/03_cognitiontest/03_biologicalsamples-01.png', auto_pdf = TRUE, dpi = NULL)
```

The orginal design was 4 animals per treament and 3 hippocampal sub regions per animals, which would give 24 samples. After excluding compromized samples, the final sample sizes are: 

```{r viewcoldata, echo=FALSE,message=FALSE, warning=FALSE}
colData %>% select(Treatment,Region)  %>%  summary()
```

### Differential gene expresssion analysis

Raw reads were downloaded from the Amazon cloud server to the Stampede Cluster at the Texas Advanced Computing Facility for processing and analysis. RNA quality was checked using the bioinformatic program FASTQC (citation). Low quality reads and adapter sequences were removed using the program Cutadapt (Martin, 2011). Kallisto was use for fast read mapping and counting (Bray et al., 2016). Transcript from a single gene were combined into a count total for each gene. In the end, we meausred the expression of 22,485 genes in 22 samples.

```{r viewscountData, echo=FALSE,message=FALSE, warning=FALSE}
dim(countData)
```

We used DESeq2 (Love et al., 2014) for gene expression normalization and quantification using the following experimental design: `Treatment + Region + Treatment * Region`. Genes with less than 2 counts across all samples were filtered, leaving us with `dim(rld)` genes for analysis of differntial expression.

```{r, include=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Treatment + Region + Treatment * Region )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## filter genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) # log transform data
```

```{r, echo=TRUE, message=TRUE,comment=FALSE, warning=FALSE}
dim(rld)
```

We identified 423 genes were differentially expressed between the yoked control and cognitively trained animals, 3485 genes that were differentially expressed across subfields, and 324 showed an interaction at FDR p < 0.05 (Fig. 4B). We see a large effect of brain region on gene expression, with 20% of detectable genes begin differentially expressed between one or more brain-region comparisons (3485 differentially expressed genes /17320 measured genes). This is an order of magnitude greater than the 2% of the transcriptome that changed in response to learning (423 DEGs /17320 genes measured).

```{r signficiantgenes, include=FALSE}
## DEG by contrasts
source("resvalsfunction.R")
contrast1 <- resvals(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.05)
contrast2 <- resvals(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.05)
contrast3 <- resvals(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.05)
contrast4 <- resvals(contrastvector = c('Treatment', 'trained', 'yoked'), mypval = 0.05)
```

```{r VennDiagramPadj, echo=FALSE}
#create a new DF with the gene counts
rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast1, contrast2, contrast3, contrast4)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]


# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[2] <0.05 & !is.na(rldpvals[2]),])
venn2 <- row.names(rldpvals[rldpvals[4] <0.05 & !is.na(rldpvals[4]),])
venn3 <- row.names(rldpvals[rldpvals[6] <0.05 & !is.na(rldpvals[6]),])
venn4 <- row.names(rldpvals[rldpvals[8] <0.05 & !is.na(rldpvals[8]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)

## check order for correctness
candidates <- list("Region" = venn123, "Treatment" = venn4)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)

# save files for big venn diagram
write(venn123, "../results/03_cognition_venn123.txt")
write(venn4, "../results/03_cognition_venn4.txt")
write(venn1, "../results/03_cognition_venn1.txt")
```

Hierarchical clustering of the differentially expressed genes separates samples by both subfield and treatment (Fig. 4C). 

Then, we visuazlied the data as a heatmap showing the relative log fold change of gene expression across samples. Genes were filtered for a minimimum adjust p value < 0.05 in any two-way contrast. The row mean for each gene was subtracted for the raw value to allow for analysis of fold change rather than raw magnitudes. The samples cluster primarily by brain region with some small treatment-driven. 

```{r HeatmapPadj, echo=FALSE, message=TRUE, comment=FALSE, warning=FALSE}
## Any padj <0.05
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

DEGes$padjmin <- with(DEGes, pmin(padjTreatmenttrainedyoked, padjRegionCA1DG ,padjRegionCA3DG, padjRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.05)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)


# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsbehavior
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Treatment", "Region")])
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))


pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 14, fontsize_row = 7, 
         treeheight_row = 0, treeheight_col = 25,
         cellwidth = 12, 
         border_color = "grey60" ,
         color = colorpalette,
         clustering_distance_cols="correlation" ,
         breaks=myBreaks,
         clustering_method="average"
         )

# for adobe
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 8, 
         width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         cellwidth = 6, 
         filename = "../figures/03_cognitiontest/HeatmapPadj-1.pdf",
         clustering_distance_cols="correlation" ,
         breaks=myBreaks,
         clustering_method="average"
         )
```

```{r pvclust}
result <- pvclust(DEGes, method.dist="cor", method.hclust="average", nboot=1000)
plot(result)
```

A principal component analysis of all gene expression data revealed that brain region explains 75% of the variance in the data (Fig. 4D).  PC1 accounts for 56% of the variance and distinguishes DG from not-DG samples (ANOVA for PC1 ~ Region: F2,19= 226.1; p < 0.001). A post hoc Tukey test showed that DG samples are significantly different from both CA1 and CA3 samples (CA1-DG, p < 0.001; CA3-DG, p < 0.001; CA1-CA3, p = 0.23).  PC2 accounts for 19% of the variance and distinguishes the three subfields (PC2 ~ Region ANOVA: F2,19= 255.3; p < 0.001; Tukey test, p < 0.001for all three comparisons). PC3 are PC4 are influenced by variation due to cognitive training (PC3 ~ Treatment ANOVA: F1,20=7.451; p = 0.012, PC4 ~ Treatment ANOVA: F1,20=10.11; p = 0.0047). An analysis of Gene Ontology identified 91 GO terms at 10% FDR significant. Among the top 10 are glutamate signaling and membrane transport systems and a downregulation of oxidoreductase and ribosomal activity (Fig. 2C). 


```{r PCA, include=FALSE}
source("DESeqPCAfunction.R")
source("figureoptions.R")

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Treatment", "Region"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

pcadata$Treatment <- factor(pcadata$Treatment, levels = c("yoked", "trained"))


# PC1 vs PC2 for adobe
myplot <- plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)
pdf(file="../figures/03_cognitiontest/PCA-1.pdf", width=4.5, height=3)
plot(myplot)
dev.off()
```


```{r PCA21, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
## PC2 vs PC1
plotPC2PC1(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)
```

```{r statistics}
## statistics
aov1R <- aov(PC1 ~ Region, data=pcadata)
summary(aov1R) 
TukeyHSD(aov1R, which = "Region") 

aov2R <- aov(PC2 ~ Region, data=pcadata)
summary(aov2R) 
TukeyHSD(aov2R, which = "Region") 

aov3R <- aov(PC3 ~ Region, data=pcadata)
summary(aov3R) 
TukeyHSD(aov3R, which = "Region") 

aov4R <- aov(PC3 ~ Region, data=pcadata)
summary(aov4R) 
TukeyHSD(aov4R, which = "Region") 

aov5R <- aov(PC5 ~ Region, data=pcadata)
summary(aov5R) 
TukeyHSD(aov5R, which = "Region") 

aov6R <- aov(PC6 ~ Region, data=pcadata)
summary(aov6R) 
TukeyHSD(aov6R, which = "Region") 


aov1T <- aov(PC1 ~ Treatment, data=pcadata)
summary(aov1T) 

aov2T <- aov(PC2 ~ Treatment, data=pcadata)
summary(aov2T) 

aov3T <- aov(PC3 ~ Treatment, data=pcadata)
summary(aov3T) 
 
aov4T <- aov(PC4 ~ Treatment, data=pcadata)
summary(aov4T) 

aov5T <- aov(PC5 ~ Treatment, data=pcadata)
summary(aov5T) 
 
aov6T <- aov(PC6 ~ Treatment, data=pcadata)
summary(aov6T) 
```

The gene expression data were exported to csv files for importing into the GOMMU analysis package for subsequent analysis.

```{r GOsetup}
# from https://github.com/rachelwright8/Ahya-White-Syndromes/blob/master/deseq2_Ahya.R
res <- results(dds, contrast=c('Treatment', 'trained', 'yoked'), independentFiltering = FALSE)
table(res$padj<0.05)

logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP <- as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign

write.csv(logs, file = "./06_GO_MWU/03_behavior_GOpvals.csv", row.names = F)
```



Supplementary figures showing the distibution of pvalues. 

```{r histogram}
myhistogram(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.05)
myhistogram(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.05)
myhistogram(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.05)
myhistogram(contrastvector = c('Treatment', 'trained', 'yoked'), mypval = 0.05)
```


Supplementary figures of showing PC3 and PC4 contrasted against PC2.)
```{r PCA34}
## PC2 vs PC3
A <- plotPC2PC3(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)

B <- plotPC2PC4(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)

plot_grid(A, B, rel_widths = c(2.25,3))
```


Here is the corresponding Adobe Illustrator file that combines many of the above plots. 
```{r summary image, echo=FALSE, warning=FALSE}
include_graphics('../figures/03_cognitiontest/03_cognitiontest-01.png', auto_pdf = TRUE, dpi = NULL)
```