---
title: "03_cognitiontest.Rmd"
author: "Rayna M Harris"
date: "March 6, 2017"
output: md_document
---
## All together now

Combining the two previous analyses

```{r setup, message=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
#library(magrittr)
#library(tidyverse)
#library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(cowplot)
library(RColorBrewer)
library(dplyr)
library(plyr)
library(ggplot2)
library(colorRamps)
#library(edgeR)
# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/03_cognitiontest/')
```

```{r ImportData, message=FALSE, warning=FALSE}
# this starts with data genearated from code described in KallistoGather.Rmd
colData <- read.csv('../data/BehaviorSlimColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/BehaviorSlimCountData.csv', check.names = F, row.names = 1)

#rename revalue things
colData$Group <- plyr::revalue(colData$Group, c("consistent"="avoidable"))
colData$Group <- plyr::revalue(colData$Group, c("control"="unavoidable"))

colData <- rename(colData, c("Group"="Treatment"))
colData$Treatment <- factor(colData$Treatment, levels = c("unavoidable", "avoidable"))

```

```{r DifferentialGeneExpressionAnalysis, echo=TRUE, message=TRUE,comment=FALSE, warning=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Treatment + Region + Treatment * Region )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## filter genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis
dds

## log transformed data
rld <- rlog(dds, blind=FALSE)
```

PCA

```{r PCA, echo=TRUE, message=FALSE, comment=FALSE, warning=FALSE}
source("DESeqPCAfunction.R")
source("figureoptions.R")

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Treatment", "Region"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

pcadata$Treatment <- factor(pcadata$Treatment, levels = c("unavoidable", "avoidable"))

## PC1 vs PC2
plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)

# PC1 vs PC2 for adobe
myplot <- plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)
pdf(file="../figures/03_cognitiontest/PCA-1.pdf", width=4.5, height=3)
plot(myplot)
dev.off()

## PC2 vs PC3
plotPC2PC3(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)

plotPC3PC4(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)

## statistics
aov1 <- aov(PC1 ~ Region, data=pcadata)
summary(aov1) 
TukeyHSD(aov1, which = "Region") 

aov2 <- aov(PC2 ~ Region, data=pcadata)
summary(aov2) 
TukeyHSD(aov2, which = "Region") 

aov5 <- aov(PC3 ~ Region, data=pcadata)
summary(aov5) 
TukeyHSD(aov5, which = "Region") 

aov3 <- aov(PC1 ~ Treatment, data=pcadata)
summary(aov3) 
TukeyHSD(aov3, which = "Treatment")

aov4 <- aov(PC2 ~ Treatment, data=pcadata)
summary(aov4) 
TukeyHSD(aov4, which = "Treatment") 

aov6 <- aov(PC3 ~ Treatment, data=pcadata)
summary(aov6) 
TukeyHSD(aov6, which = "Treatment") 

aov7 <- aov(PC4 ~ Treatment, data=pcadata)
summary(aov7) 
TukeyHSD(aov7, which = "Treatment") 

lm1 <- lm(PC1~Region*Treatment, data=pcadata)
summary(lm1)
anova(lm1) 

lm2 <- lm(PC2~Region*Treatment, data=pcadata)
summary(lm2)
anova(lm2)
```


```{r signficiantgenes}
## DEG by contrasts
source("resvalsfunction.R")
contrast1 <- resvals(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.1)
contrast2 <- resvals(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.1)
contrast3 <- resvals(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.1)
contrast4 <- resvals(contrastvector = c('Treatment', 'avoidable', 'unavoidable'), mypval = 0.1)
```


Now, we can view a histogram of the distribution 

```{r histogram, echo=FALSE}
myhistogram(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.1)
myhistogram(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.1)
myhistogram(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.1)
myhistogram(contrastvector = c('Treatment', 'avoidable', 'unavoidable'), mypval = 0.1)

```


This Venn Diagram sthe overlap of differentailly expression genes by Region and method. This shows all genes with *adjusted* pvalue <0.1.


```{r VennDiagramPadj, echo=FALSE}
#create a new DF with the gene counts
rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast1, contrast2, contrast3, contrast4)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]


# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[2] <0.1 & !is.na(rldpvals[2]),])
venn2 <- row.names(rldpvals[rldpvals[4] <0.1 & !is.na(rldpvals[4]),])
venn3 <- row.names(rldpvals[rldpvals[6] <0.1 & !is.na(rldpvals[6]),])
venn4 <- row.names(rldpvals[rldpvals[8] <0.1 & !is.na(rldpvals[8]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)

## check order for correctness
candidates <- list("Region" = venn123, "Treatment" = venn4)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)

# save files for big venn diagram
write(venn123, "../results/03_cognition_venn123.txt")
write(venn4, "../results/03_cognition_venn4.txt")
write(venn1, "../results/03_cognition_venn1.txt")

```

Heatmaps

```{r HeatmapPadj, echo=TRUE, message=TRUE, comment=FALSE, warning=FALSE}
## Any padj <0.1
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

DEGes$padjmin <- with(DEGes, pmin(padjTreatmentavoidableunavoidable, padjRegionCA1DG ,padjRegionCA3DG, padjRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)


# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsbehavior
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Treatment", "Region")])
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))


pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         fontsize = 12, fontsize_row = 7, 
         cellwidth = 10, 
         border_color = "grey60" ,
         color = colorpalette,
         clustering_distance_cols="correlation" ,
         breaks=myBreaks,
         clustering_method="average"
         )

# for adobe
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         cellwidth = 11, 
         filename = "../figures/03_cognitiontest/HeatmapPadj-1.pdf",
         clustering_distance_cols="correlation" ,
         breaks=myBreaks,
         clustering_method="average"
         )

```

```{r HeatmapPvalue, message=FALSE, warning=FALSE}
## Any pvalue <0.1
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

DEGes$pvalmin <- with(DEGes, pmin(pvalTreatmentavoidableunavoidable, pvalRegionCA1DG ,pvalRegionCA3DG, pvalRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(pvalmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)


# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsbehavior
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Treatment", "Region")])

paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

# for markdown file
pheatmap(DEGes, show_colnames=T, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         border_color = "grey60" ,
         color = colorpalette,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

# for adobe
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         cellwidth = 8, 
         filename = "../figures/03_cognitiontest/HeatmapPadj-1.pdf",
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )
```

```{r GOsetup, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE }
# from https://github.com/rachelwright8/Ahya-White-Syndromes/blob/master/deseq2_Ahya.R

res <- results(dds, contrast=c('Treatment', 'avoidable', 'unavoidable'))
table(res$padj<0.1)
table(res$pvalue<0.05)
head(res)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP <- as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign

write.csv(logs, file = "./06_GO_MWU/03_behavior_GOpvals.csv", row.names = F)
```

```{r numshocks, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE }
## now, to show how many shocks each animal received.
avoidableshock <- read.csv("../data/avoidableshock.csv", header = T)
avoidableshock$Treatment <- factor(avoidableshock$Treatment, levels = c("unavoidable", "avoidable"))


A <- avoidableshock %>% 
  filter(TrainSessionCombo.y != "Hab") %>% droplevels() %>%
  ggplot(aes(x=TrainSessionCombo.y, y=NumShock.y, color=Treatment)) + 
  geom_point(size = 2 ,position = position_jitter(width = 0.3, height = 0)) +
  theme_cowplot(font_size = 16, line_size = 1) + 
  background_grid(major = "xy", minor = "y") + 
  scale_x_discrete(labels=c("T1" = "1", "T2" = "2",
                            "T3" = "3", "T4_C1" = "4",
                            "T5_C2" = "5", "T6_C3" = "6")) + 
  labs(list(x = "Day", y = "Number of Shocks")) +
  ylim(0, 15) + 
  scale_color_manual(values=c("#969696", "#525252")) +
  theme(legend.position="bottom",
        legend.title=element_blank())

B <- avoidableshock %>% 
  ggplot(aes(x= as.numeric(TrainSessionCombo.y), y=pTimeTarget, color=Treatment)) + 
  geom_point(size = 2 ,position = position_jitter(width = 0.3, height = 0)) +
  stat_smooth(alpha=0.2, size=2)  +
  theme_cowplot(font_size = 16, line_size = 1) + 
  background_grid(major = "xy", minor = "none") + 
  scale_y_continuous(name="% Time in North Quadrant") + 
  scale_x_continuous(name = "Day", 
                     breaks = c(1, 2, 3, 4, 5, 6, 7),
                     labels=c("1" = "0", "2" = "1", "3" = "2", 
                              "4" = "3", "5" = "4", "6" = "5",
                              "7" = "6")) + 
  scale_color_manual(values=c("#969696", "#525252")) +
  theme(legend.position="bottom",
        legend.title=element_blank())


plot_grid(A, B, rel_widths = c(1,1))

# PC1 vs PC2 for adobe
myplot <- plot_grid(A, B, rel_widths = c(1,1))
pdf(file="../figures/03_cognitiontest/behavior.pdf", width= 6.5, height = 4)
plot(myplot)
dev.off()
```

```{r ephys, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE }
## now, to show how many shocks each animal received.
inputoutput <- read.csv("../data/inputoutput.csv", header = T)
LTP <- read.csv("../data/LTP.csv", header = T)


inputoutput$Treatment <- factor(inputoutput$Treatment, levels = c("unavoidable", "avoidable"))
LTP$Treatment <- factor(LTP$Treatment, levels = c("unavoidable", "avoidable"))

pd <- position_dodge(0) # move them .05 to the left and right

ggplot(inputoutput, aes(x=Stimulus, y=fEPSP, colour=Treatment)) +
    geom_errorbar(aes(ymin=fEPSP-SE, ymax=fEPSP+SE), width=.1) +
    #geom_line() +
    geom_point(size = 3, position=pd) +
  scale_color_manual(values=c("#969696", "#525252")) +
  scale_x_continuous(name="Stimulation Intensity (V)") +
    scale_y_continuous(name="fEPSP slope (mV/ms)", trans = "reverse") +
  theme_cowplot(font_size = 16, line_size = 1) +
  theme(legend.position="bottom",
        legend.title=element_blank()) + 
  background_grid(major = "xy", minor = "none") 

ggplot(LTP, aes(x=Time, y=fEPSPslope, colour=Treatment)) + 
    geom_errorbar(aes(ymin=fEPSPslope-SE, ymax=fEPSPslope+SE), width=.1, position=pd) +
    #geom_line() +
    geom_point(size = 2, position=pd) +
  scale_color_manual(values=c("#969696", "#525252")) +
  scale_x_continuous(limits = c(-15, 30), name="Time to Stimulation (min)") +
    scale_y_continuous(name="fEPSP slope (% to baseline)") +
  theme_cowplot(font_size = 16, line_size = 1) +
  theme(legend.position="bottom",
        legend.title=element_blank()) + 
  background_grid(major = "xy", minor = "none") 
```


```{r pvclust, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE }
library(pvclust)
# clustering just the degs
result <- pvclust(DEGes, method.dist="cor", method.hclust="average", nboot=1000)
plot(result)
```