---
title: "03_DissocationStressTest.Rmd"
author: "Rayna M Harris"
date: "March 6, 2017"
output: md_document
---
## All together now

Combining the two previous analyses

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(cowplot)
library(RColorBrewer)
library(ggcorrplot)
library(dplyr)
library(plyr)
library(ggplot2)
library(colorRamps)

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/03_dissociationstresstest/')
```

```{r ImportData, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
# this starts with data genearated from code described in KallistoGather.Rmd
colData <- read.csv('../data/DissociationColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/DissociationCountData.csv', check.names = F, row.names = 1)
colData$Group <- plyr::revalue(colData$Group, c("control"="stressed"))

colData$NewGroup <- as.factor(paste(colData$Group, colData$Method,sep="_"))
```

```{r DifferentialGeneExpressionAnalysis, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ NewGroup + Region + NewGroup * Region )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## filter genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis
dds

# general deseq
res <- results(dds, independentFiltering = F)

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- getVarianceStabilizedData(dds)
```

PCA

```{r PCA, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
source("DESeqPCAfunction.R")
source("figureoptions.R")

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("NewGroup", "Region"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$NewGroup, shapename = "Group", colorvalues = colorvalRegion)


```

