---
title: "meta123"
author: "Rayna M Harris"
date: "April 6 2017"
output: html_document
---

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(plyr)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(edgeR)
library(colorRamps)

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/07_meta123/')
```

```{r ImportData, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
# this starts with data genearated from code described in ../../BehavEphysRNAseq/markdownfiles/02a_tidysamples.R and ../../BehavEphysRNAseq/markdownfiles/02b_KallistoGather.Rmd
colData1 <- read.csv('../data/DissociationColData.csv')
rownames(colData1) <- colData1$RNAseqID
countData1 <-  read.csv('../data/DissociationCountData.csv', check.names = F, row.names = 1)

colData2 <- read.csv('../data/BehaviorSlimColData.csv')
rownames(colData2) <- colData2$RNAseqID
countData2 <-  read.csv('../data/BehaviorSlimCountData.csv', check.names = F, row.names = 1)

colData <- rbind(colData1, colData2)
countData <- cbind(countData1, countData2)
```


```{r SubsetData, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
## Subset to just look homogenized and dissociated samples
colData <- colData %>%
  filter(RNAseqID != "101-DG-3") %>% droplevels()
savecols <- as.character(colData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols)) # keep good samples


## rename and relevel things
colData <- rename(colData, c("Group"="Treatment"))
#colData$Treatment <- revalue(colData$Treatment, c("control"="homogenized"))
#colData$Treatment <- factor(colData$Treatment, levels = c("homogenized", "dissociated"))
```

Here is a brief overview of the samples being compared.

```{r sampleinfo}
str(colData)
summary(colData)
```


```{r DEG, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Treatment + Region + Treatment * Region )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## filter genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis
dds

## log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- getVarianceStabilizedData(dds)
```

Some read statistics
```{r edgeR, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE }
counts <- countData
dim( counts )
colSums( counts ) / 1e06  # in millions of reads
table( rowSums( counts ) )[ 1:30 ] # Number of genes with low counts

rowsum <- as.data.frame(colSums( counts ) / 1e06 )
names(rowsum)[1] <- "millioncounts"
rowsum$sample <- row.names(rowsum)
summary(rowsum)

ggplot(rowsum, aes(x=millioncounts)) + 
  geom_histogram(bins = 20, colour = "black", fill = "darkgrey") +
  theme_classic() +
  scale_x_continuous(name = "Millions of Gene Counts per Sample") +
  scale_y_continuous(name = "Number of Samples")
```

This PCA gives an overview of the variability between samples using the a large matrix of log transformed gene expression data. You can see that the bigges difference is between DG punches and the CA1 and CA3 punches. CA1 and CA3 samples have similar transcriptomes. The homogenized CA1 samples have the most similar transcriptonal profiles as evidenced by their tight clustering. 

```{r PCA, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
source("DESeqPCAfunction.R")
source("figureoptions.R")

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Region", "Treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))

## for markdown
plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", colorvalues = colorvalRegion, aesshape = pcadata$Treatment, shapename = "Treatment")

plotPC2PC3(aescolor = pcadata$Region, colorname = "Region", colorvalues = colorvalRegion, aesshape = pcadata$Treatment, shapename = "Treatment")

# for adobe
myplot <- plotPC1PC2(aescolor = pcadata$Region, colorname = "Region", aesshape = pcadata$Treatment, shapename = "Treatment", colorvalues = colorvalRegion)
pdf(file="../figures/07_meta123/PCA-1.pdf", width=4.5, height=3)
plot(myplot)
dev.off()
```

Now, we can calulate the number of significant genes by contrast by contrast. The first number displayed is not corrected for mutiple hypothesis testing but the second one is.

```{r signficiantgenes}
## DEG by contrasts
source("resvalsfunction.R")
contrast1 <- resvals(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.1)
contrast2 <- resvals(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.1)
contrast3 <- resvals(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.1)

```


Now, we can view a histogram of the distribution 

```{r histogram, echo=FALSE}
source("resvalsfunction.R")
myhistogram(contrastvector = c('Region', 'CA1', 'DG'), mypval = 0.1)
myhistogram(contrastvector = c('Region', 'CA3', 'DG'), mypval = 0.1)
myhistogram(contrastvector = c('Region', 'CA1', 'CA3'), mypval = 0.1)

```


This Venn Diagram sthe overlap of differentailly expression genes by Region and Treatment. This shows all genes with *adjusted* pvalue <0.1.

```{r VennDiagramPadj, echo=FALSE}
#create a new DF with the gene counts
rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast1, contrast2, contrast3)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]


# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[2] <0.1 & !is.na(rldpvals[2]),])
venn2 <- row.names(rldpvals[rldpvals[4] <0.1 & !is.na(rldpvals[4]),])
venn3 <- row.names(rldpvals[rldpvals[6] <0.1 & !is.na(rldpvals[6]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)

```


#heatmaps

```{r HeatmapPadj, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
## Any padj <0.1
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe

DEGes$padjmin <- with(DEGes, pmin(padjRegionCA1DG ,padjRegionCA3DG, padjRegionCA1CA3 )) # put the min pvalue in a new column
DEGes <- DEGes %>% filter(padjmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

# setting color options
source("figureoptions.R")
ann_colors <- ann_colorsmeta
colorpalette <- cembrowskicolors
df <- as.data.frame(colData(dds)[,c("Treatment", "Region")])
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))


pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         #fontsize = 12, fontsize_row = 7, 
         #cellwidth=20,
         treeheight_row = 0, treeheight_col = 25,
         border_color = "grey60" ,
         color = colorpalette,
         breaks=myBreaks,
         clustering_distance_cols="correlation"
         )
```

This is a data validation check plot. Here, I'm showing how many millions of reads were present in each sample. On average, each sample had 5 million reads, but the range was from 0.8 to 10 millino reads. 

```{r edgeR2, echo=FALSE, message=FALSE,comment=FALSE, warning=FALSE }
library(edgeR)
counts <- countData
dim( counts )
colSums( counts ) / 1e06  # in millions of gene counts
table( rowSums( counts ) )[ 1:30 ] # Number of genes with low counts
```
