---
title: "GO_MWU.Rmd"
author: "Rayna M Harris"
date: January 7, 2017
output:
  md_document:
  variant: markdown_github
---

GO_MWU uses continuous measure of significance (such as fold-change or -log(p-value) ) to identify GO categories that are significantly enriches with either up- or down-regulated genes. The advantage - no need to impose arbitrary significance cutoff.

If the measure is binary (0 or 1) the script will perform a typical "GO enrichment" analysis based Fisher's exact test: it will show GO categories over-represented among the genes that have 1 as their measure. 

On the plot, different fonts are used to indicate significance and color indicates enrichment with either up (red) or down (blue) regulated genes. No colors are shown for binary measure analysis.

The tree on the plot is hierarchical clustering of GO categories based on shared genes. Categories with no branch length between them are subsets of each other.

The fraction next to GO category name indicates the fracton of "good" genes in it; "good" genes being the ones exceeding the arbitrary absValue cutoff (option in gomwuPlot). For Fisher's based test, specify absValue=0.5. This value does not affect statistics and is used for plotting only.

Stretch the plot manually to match tree to text

Mikhail V. Matz, UT Austin, February 2015; matz@utexas.edu

################################################################

```{r setup}
library(ape)
source("gomwu.functions.R")

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../../figures/06_GO_MMU/')
```

## From Experitment 1: Dissociation Test Molecular Function (MF)

```{r 01-dissociationMF}
# input files
input="01_dissociation_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="MF" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25)  

# Data viz
gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

# Data viz
gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.001, 
	level2=0.0005, 
	level3=0.0001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)
```

## From Experitment 1: Dissociation Test Cellular Component (CC)


```{r 01-dissociationCC, echo=FALSE}
# input files
input="01_dissociation_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="CC" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25 
)
<<<<<<< HEAD

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.000001, 
	level2=0.0000005, 
	level3=0.0000001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)
```

## From Experitment 1: Biological Process (BP)


```{r 01-dissociationBP, echo=FALSE}
# input files
input="01_dissociation_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="BP" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25 
)
gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
=======
gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
>>>>>>> 06a9ef068c8c11ec58b2295f4e31fb03700f4607
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.5,    
	treeHeight=0.5, #
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.01, 
	level2=0.005, 
	level3=0.001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.001, 
	level2=0.0005, 
	level3=0.0001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)
```


## From Experitment 2: Stress Molecular Function (MF)

```{r 02_stressMF, echo=FALSE}
# input files
input="02_stress_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="MF" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25 
)

# 0  GO terms at 10% FDR 
# therefore, no plots

```

## From Experitment 2: Stress Cellular Component (CC)


```{r 02_stressCC, echo=FALSE}
# input files
input="02_stress_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="CC" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25 
)

# 0  GO terms at 10% FDR 
# therefore, no plots

```


## From Experitment 3: Cognition Cellular Component (CC)

```{r 03_behaviorCC, echo=FALSE}
# input files
input="03_behavior_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="CC" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25 
)
# do not continue if the printout shows that no GO terms pass 10% FDR.

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.5,    
	treeHeight=0.5, #
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.01, 
	level2=0.005, 
	level3=0.001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.001, 
	level2=0.0005, 
	level3=0.0001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)
<<<<<<< HEAD

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.0001, 
	level2=0.00005, 
	level3=0.00001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)
=======
>>>>>>> 06a9ef068c8c11ec58b2295f4e31fb03700f4607
```


## From Experitment 3: Cognition Molecular Function (MF)

```{r 03_behaviorMF, echo=FALSE}
# input files
input="03_behavior_GOpvals.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="MF" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.5,    
	treeHeight=0.5, #
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.01, 
	level2=0.005, 
	level3=0.001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=-log(0.05,10),  
	level1=0.001, 
	level2=0.0005, 
	level3=0.0001, 
	txtsize=1.4,    
	treeHeight=0.5, 
  colors=c("dodgerblue2","firebrick1","skyblue","lightcoral") 
)
```


## Now for Presence/Absence GO analysis
- First, I exported the list of overlapping genes from the previous 05_combo.Rmd file
- Now, I'll use some joining majic to create a file with all the genes in the study and a 0 or 1 based on whether or not they were in the list of genes

```{r dplyr}
library(dplyr)

#list of all genes. I only care about column 1
allgenes <- read.csv("01_dissociation_GOpvals.csv", header = T)
allgenes <- select(allgenes, gene)

# first, identify genes differentially expressed by region in all experiments
intersection <- read.csv("05_combo_intersection.csv", header=T)
intersection$PresAbs <- 1
intersection <- full_join(allgenes, intersection)
intersection <- intersection %>%
   mutate(PresAbs = replace(PresAbs,is.na(PresAbs),0))
str(intersection)

write.csv(intersection, "./05_combo_intersection_allgenes.csv", row.names = F)

```

## The intersection: Cellular component

```{r 05-intersectionCC, echo=FALSE}
# input files
input="05_combo_intersection_allgenes.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="CC" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25,
	Alternative="g"
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=0.001,   
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.5,    
	treeHeight=0.5 
)
```

## The intersection: Molecular Function

```{r 05-intersectionMF, echo=FALSE}
# input files
input="05_combo_intersection_allgenes.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="MF" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25,
	Alternative="g"
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=0.001,   
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.5,    
	treeHeight=0.5
)
```
<<<<<<< HEAD

## The intersection: Biological Process

```{r 05-intersectionBP, echo=FALSE}
# input files
input="05_combo_intersection_allgenes.csv" 
goAnnotations="goAnnotations.tab" 
goDatabase="go.obo" 
goDivision="BP" # either MF, or BP, or CC

# Calculating stats
gomwuStats(input, goDatabase, goAnnotations, goDivision,
	perlPath="perl", 
	largest=0.1,  
	smallest=5,   
	clusterCutHeight=0.25,
	Alternative="g"
)

gomwuPlot(input,goAnnotations,goDivision,
	absValue=0.001,   
	level1=0.1, 
	level2=0.05, 
	level3=0.01, 
	txtsize=1.5,    
	treeHeight=0.5 
)
```
=======
>>>>>>> 06a9ef068c8c11ec58b2295f4e31fb03700f4607
