---
title: "01_deseq2.Rmd"
output: md_document
---
### Identifying the effects of cellular dissociation on hippocampal transcriptomes

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#source("http://www.bioconductor.org/biocLite.R")
#biocLite("edgeR")
library(DESeq2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(VennDiagram)
library(genefilter)
library(pheatmap)
library(cowplot)
library(RColorBrewer)
library(dplyr)
library(plyr)
library(ggplot2)
library(car)
library(edgeR)

# user defined functions and options
source("resvalsfunction.R") 
source("DESeqPCAfunction.R")

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/01_deseq2/',
                      message=FALSE, warning=FALSE)
```

The sample and count information for this part is found in `../results/GSE99765_DissociationColData.csv` and `../results/GSE99765_DissociationCountData.csv`. You can also download these two files from [GEO GSE99765](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99765). 

```{r ImportData, message=FALSE, warning=FALSE}
colData <- read.csv('../results/GSE99765_DissociationColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../results/GSE99765_DissociationCountData.csv', check.names = F, row.names = 1)
```

Do a little data cleaning and calculate sample size and number of genes measured. 

```{r viewcoldata}
# rename column and samples. 
colData <- rename(colData, c("Region"="Subfield"))
colData$Treatment <- revalue(colData$Treatment, c("control"="HOMO", "dissociated"="DISS"))

# calculate samples size and number of genes for which we have expression data
table(colData$Treatment,colData$Subfield) 
dim(countData)
```

I used DESeq2 (Love et al., 2014) for gene expression normalization and quantification using the following experimental design: `Treatment + Subfield + Treatment * Subfield`. Genes with less than 2 counts across all samples were filtered, leaving us with `dim(rld)` number of genes for analysis of differntial expression.

```{r DEG}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Treatment + Subfield + Treatment * Subfield )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## pre-filter genes 
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data
dim(rld) #print total genes analyzed
vsd <- vst(dds, blind=FALSE) # variance stabilized
head(assay(rld), 3)
head(assay(vsd), 3)

# save results 
write.csv(assay(vsd), "../results/vsd.csv")
write.csv(assay(rld), "../results/rld.csv")
```

We identified 344 genes that were differentially expressed between the homogenized and dissociated samples at FDR p-value < 0.1. 


```{r signficiantgenes}
## DEG by contrasts at 0.1 pvalue
contrast1 <- resvals(contrastvector = c('Subfield', 'CA1', 'DG'), mypval = 0.1) #484
contrast2 <- resvals(contrastvector = c('Subfield', 'CA3', 'DG'), mypval = 0.1) #98
contrast3 <- resvals(contrastvector = c('Subfield', 'CA1', 'CA3'), mypval = 0.1) #18
contrast4 <- resvals(contrastvector = c('Treatment', 'DISS', 'HOMO'), mypval = 0.1) #344

# % transcrptiome altered by treatment
344/16709 * 100
```


A hierarchical clustering analysis of all differentially expressed genes does not give rise to distinct clusters that are separated by subfield or method; however, when examining the control, homogenized samples alone (identified with light grey boxes), the three subfields form distinct clusters, while the dissociated samples do not cluster by subfield. 


```{r HeatmapPadj}
contrast4 <- resvals(contrastvector = c('Treatment', 'DISS', 'HOMO'), mypval = 0.01)

DEGs <- assay(rld)
DEGs <- cbind(DEGs, contrast4)
DEGs <- as.data.frame(DEGs) # convert matrix to dataframe
DEGs$rownames <- rownames(DEGs)  # add the rownames to the dataframe

DEGs$padjmin <- with(DEGs, pmin(padjTreatmentDISSHOMO)) # put the min pvalue in a new column

write.csv(as.data.frame(DEGs), "../results/heatmap_DEGs.csv", row.names = F)
write.csv(colData, "../results/heatmap_colData.csv", row.names = F) 
```

## Volcano Plots

Craete new data frames that include fold change, pvalue, and a column describing the direction for differential gene expression. This "direction" will be used to color code the dots on the volcano plot. Will also save a list of DEGs at the end. 

```{r volcanotreatment}
res <- results(dds, contrast =c('Treatment', 'DISS', 'HOMO'), independentFiltering = T, alpha = 0.01)
summary(res)

res <- results(dds, contrast =c('Treatment', 'DISS', 'HOMO'), independentFiltering = T, alpha = 0.1)
summary(res)

288+56 # tolal number of DEGs = 344
(344/16709)*100 # percent of DEGs out of total measured

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange,
                   padj = res$padj)
data <- na.omit(data)
data <- data %>%
    mutate(direction = ifelse(data$lfc > 1 & data$pvalue > 1, 
                        yes = "DISS", 
                        no = ifelse(data$lfc < -1 & data$pvalue > 1, 
                                    yes = "HOMO", 
                                    no = "neither")))
data$direction <- as.factor(data$direction)
summary(data)
# note, there are fewer DEGs now because they have been filtered for higher than lfc 1.5

write.csv(data, "../results/volcanoTreatment.csv")

# save the list of just DEGs with their pvalu and lfc and direction
dissocDEGs <- data %>%
  filter(direction != "neither")
dissocDEGs <- dissocDEGs[order(dissocDEGs$padj),]
dissocDEGs$pvalue <- NULL

dissocDEGs$padj <-formatC(dissocDEGs$padj, format = "e", digits = 2) # round to scientific
dissocDEGs <- dissocDEGs %>% mutate_if(is.numeric, ~round(., 2)) # round to 2 dec

head(dissocDEGs)

write.csv(dissocDEGs, "../results/dissociationDEGs.csv", row.names = F)
```

```{r volcanosubfield}
res <- results(dds, contrast =c("Subfield", "CA1", "DG"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 3)

data <- data.frame(gene = row.names(res), 
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange, 
                   padj = res$padj )
data <- na.omit(data)
data <- data %>%
  mutate(direction = ifelse(data$lfc > 1 & data$pvalue > 1, 
                        yes = "CA1", 
                        no = ifelse(data$lfc < -1 & data$pvalue > 1, 
                                    yes = "DG", 
                                    no = "neither")))

data$direction <- as.factor(data$direction)
summary(data)
write.csv(data, "../results/volcanoCA1DG.csv")

# save the list of just DEGs with their pvalu and lfc and direction
CA1DG_DEGs <- data %>%
  filter(direction != "neither")
CA1DG_DEGs <- CA1DG_DEGs[order(CA1DG_DEGs$pvalue),]

write.csv(CA1DG_DEGs, "../results/CA1DG_DEGs.csv")
```

```{r volcanodontplot}
res <- results(dds, contrast =c("Subfield", "CA3", "DG"), independentFiltering = T, alpha = 0.1)
summary(res)
res <- results(dds, contrast =c("Subfield", "CA1", "CA3"), independentFiltering = T, alpha = 0.1)
summary(res)
```

## Table 1: % of DEGs

```{r}
(222+262)/16709*100
(45+53)/16709*100
(17+1)/16709*100
(56+288)/16709*100
```


## PCA

This PCA gives an overview of the variability between samples using the a large matrix of log transformed gene expression data. You can see that the biggest difference is between DG punches and the CA1 and CA3 punches. CA1 and CA3 samples have similar transcriptomes. The control CA1 samples have the most similar transcriptonal profiles as evidenced by their tight clustering. 

```{r PCA, message=FALSE, warning=FALSE}
colorvalSubfield <- c("#7570b3", "#1b9e77", "#d95f02")
colorvalTreatment <- c("#ffffff", "#525252")

#rowVars(assay(rld))

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Subfield", "Treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
#percentVar
write.csv(pcadata, "../results/pcadata.csv")


PCA12 <- ggplot(pcadata, aes(PC1, PC2, shape = pcadata$Treatment)) + 
  geom_point(size = 5, alpha = 1, aes(color = Subfield)) +
  stat_ellipse(type = "t", aes(lty=pcadata$Treatment)) +
  scale_linetype_manual(values=c(3,1)) +
    xlab(paste0("PC1: ", percentVar[1],"% variance")) +
    ylab(paste0("PC2: ", percentVar[2],"% variance")) +
    scale_color_manual(values = colorvalSubfield) +
   theme_cowplot(font_size = 12, line_size = 0.25)  +
   theme_minimal() +
  xlim(-56,36) +
  ylim(-56,36) +
    scale_shape_manual(values=c(1, 16))  +
    theme(legend.position = "right",
          legend.key.width = unit(0.1,"mm"),
          legend.key.height = unit(0.1,"cm"),
          legend.title = element_blank(),
          legend.text = element_text(size = 8),
          panel.grid.minor=element_blank())

# thanks to https://stackoverflow.com/questions/31295382/how-to-change-the-linetype-for-ellipses-in-ggplot2-with-stat-ellipse for help with elipse

a <- ggdraw() + draw_image("../figures/00_methodsoverview/expdesign.png", scale = 1)

figure1 <- plot_grid(a, PCA12,  
                     nrow = 1, labels = c('A', 'B'), 
                     #align = 'h',
                     rel_widths = c(2,4))

figure1

pdf("../figures/figure1.pdf", width=7, height=4)
print(figure1)
dev.off()

ggsave(
  "../figures/figure1.png",
  figure1,
  width = 6,
  height = 3,
  dpi = 1200
)

```

PCA statistics

```{r}
aov1 <- aov(PC1 ~ Subfield * Treatment, data=pcadata)
summary(aov1) 
TukeyHSD(aov1, which = "Subfield") 

aov2 <- aov(PC2 ~ Subfield * Treatment, data=pcadata)
summary(aov2) 
TukeyHSD(aov2, which = "Subfield") 
```


To view a histogram of the p-value distibution for each constrast, change the Rmd file to `include=TRUE` for this chunck. 

```{r histogram, include=FALSE}
source("resvalsfunction.R")
myhistogram(contrastvector = c('Subfield', 'CA1', 'DG'), mypval = 0.05)
myhistogram(contrastvector = c('Subfield', 'CA3', 'DG'), mypval = 0.05)
myhistogram(contrastvector = c('Subfield', 'CA1', 'CA3'), mypval = 0.05)
myhistogram(contrastvector = c('Treatment', 'DISS', 'HOMO'), mypval = 0.05)
```
