---
title: "readcounts"
author: "Rayna M Harris"
date: "02/28/2019"
output: md_document
---

```{r setup}
library(tidyverse)
library(cowplot)

knitr::opts_chunk$set(fig.path = '../figures/06_readcounts/')
```

Thankfully, the Genome Sequencing and Analysis Facility maintaines friendly webarchive of sample information long after samples are processed. This included a table of reads per sample and any quality conrol data. 

## RNA concentation in each sample

```{r nanograms}
rnaconcentration <- read.csv("../results/picogreen.csv")
summary(rnaconcentration)

# format data
rnaconcentration$nanograms <- (rnaconcentration$picograms)/1000 
rnaconcentration$treatment <- factor(rnaconcentration$treatment, levels = c("HOMO", "DISS"))
rnaconcentration$subfield <- factor(rnaconcentration$subfield, levels = c("DG", "CA1", "CA3"))


a <- ggplot(rnaconcentration, 
            aes(x = treatment, y = nanograms, 
                fill = treatment, color = treatment)) + 
  geom_boxplot() +
  scale_fill_manual(values = c("white", "black")) +
  scale_color_manual(values = c("black", "grey")) +
  theme_cowplot(font_size = 12, line_size = 0.25) +
  labs(y = "concentration (ng/uL)",
       title = "RNA Concentration") +
  theme_light() +
    theme(legend.position = "none",
          panel.grid.major.x = element_blank()) 

b <- ggplot(rnaconcentration, aes(x = subfield, y = nanograms, 
                  fill = subfield)) + 
  geom_boxplot() +
  scale_fill_manual(values = c("#d95f02", "#7570b3", "#1b9e77" )) +
  theme_cowplot(font_size = 12, line_size = 0.25) +
  labs(y = "concentration (ng/uL)",
       title = "RNA Concentration") +
  theme_light() +
    theme(legend.position = "none",
          panel.grid.major.x = element_blank())


homo <- rnaconcentration %>% filter(treatment == "HOMO") 
mean(homo$nanograms)
sd(homo$nanograms)

diss <- rnaconcentration %>% filter(treatment == "DISS") 
mean(diss$nanograms)
sd(diss$nanograms)

summary(aov(nanograms ~ subfield * treatment, data=rnaconcentration))

```

## Counts per Sample

Counts per sample is provided by the GSAF. Alternatively, you can use this bash forloop to count the reads yourself. 

~~~{.bash}
for file in *R1_001.fastq.gz
do
echo $file
zcat $file | echo $((`wc -l`/4)) 
done 
~~~

A table of sample and read counts was save in `../results/readcounts.txt`.

```{r millionreads}
reads <- read.table("../results/readcounts.txt", header = T)
summary(reads)

# format data
reads$millionreads <- (reads$counts)/1000000 # to show in millions
reads$treatment <- factor(reads$treatment, levels = c("HOMO", "DISS"))
reads$subfield <- factor(reads$subfield, levels = c("DG", "CA1", "CA3"))

# stats
mean(reads$millionreads) 
sd(reads$millionreads)


c <- ggplot(reads, aes(x = treatment, y = millionreads, 
                  fill = treatment, color = treatment)) + 
  geom_boxplot() +
  scale_fill_manual(values = c("white", "black")) +
  scale_color_manual(values = c("black", "grey")) +
  theme_cowplot(font_size = 12, line_size = 0.25) +
  labs(y = "million reads per sample",
       title = "Sequencing depth") +
  theme_light() +
    theme(legend.position = "none",
          panel.grid.major.x = element_blank()) 

d <- ggplot(reads, aes(x = subfield, y = millionreads, 
                  fill = subfield)) + 
  geom_boxplot() +
  scale_fill_manual(values = c("#d95f02", "#7570b3", "#1b9e77" )) +
  theme_cowplot(font_size = 12, line_size = 0.25) +
  labs(y = "million reads per sample",
       title = "Sequencing depth") +
  theme_light() +
    theme(legend.position = "none",
          panel.grid.major.x = element_blank())




homo <- reads %>% filter(treatment == "HOMO") 
mean(homo$millionreads)
sd(homo$millionreads)


diss <- reads %>% filter(treatment == "DISS") 
mean(diss$millionreads)
sd(diss$millionreads)

summary(aov(millionreads ~ subfield * treatment, data=reads))

```

On average, my samples yielded 4.9 +/- 2.6 million reads.




```{r RNA}
plot_grid(a,b,c,d)
```



```{r}
aligned <- read.table("../results/pseudoaligned_clean.txt", header = T)

aligned$percent <- (aligned$pseudoaligned / aligned$processed) * 100
summary(aligned)

mean(aligned$percent)
sd(aligned$percent)
```

On average, 61.2% +/1 20.8% of the trimmed reads were psuedoaligned to the transcriptome.

## MultiQC

Here are the results QC before and after filtering and trimming reads

Before
![before](../figures/06_readcounts/before.png)

After
![after](../figures/06_readcounts/after.png)

