---
title: "table 2"
author: "Rayna M Harris"
date: "4/16/2018"
output: md_document
---

```{r setup, include=FALSE}
library(dplyr)
```

Import file with information about differentially expressed genes (aka `dissociation`) and the list of genes found in the reference transcriptome (aka `geneids`).

```{r}
dissociation <- read.csv("../results/01_dissociation_volcanoTreatment.csv", header = T, row.names = 1)
geneids <- read.csv("../data/geneids.csv", header = T)

```

Filter out non-significant genes, sort by p-value, rename column, round to 2 decimal places. 

```{r}
DEGs <- dissociation %>%
  filter(color != "none") %>%
  arrange((padj))
DEGs$pvalue <- NULL # drop log pvalue columns
names(DEGs)[4] <- "upregulated in"
DEGs$lfc <- signif(DEGs$lfc, digits = 2)
DEGs$padj <- signif(DEGs$padj, digits = 3)
```


```{r}
write.csv(DEGs, file = "../results/SuppTable1.csv", row.names = F)
```

Compare to Molecules implicated in hippocampal LTP

```{r}
# create list of candidate genes Sanes and Lichtman 1999 

sanesLichtman <- c("Gria1", "Gria2", 
       "Grm1", "Grm4", "Grm5", "Grm7",
       "Grin1", "Grin2a", "Grin2d", 
       "Th", "Drd1",
       "Adrb1", "Adrb2", "Adrb3",
       "Adra1a", "Adra1b", "Adra1d", 
       "Adra2a", "Adra2b", "Adra2c",
       "Oprm1", "Oprk1", "Oprd1",
       "Chrm1", "Chrm2", "Chrm3", "Chrm4", "Chrm5",
       "Chrna1", "Chrna7", "Chrna3", 
       "Chrnb1", "Chrnb2", "Chrnb3",
       "Gabra1", "Gabra2",  "Gabra3", "Gabra5", "Gabra6",
       "Gabrb1", "Gabrb2", "Gabrb3",  "Gabrr1",  "Gabbr1",
       "Cnr1", "Pnoc", "Oprl1",
       "Htr1a", "Htr1b", "Htr1f",
       "Htr2a", "Htr2c", "Htr2b",
       "Htr3a", "Htr3b", "Htr5a", "Htr5b",
       "Htr7", "Htr6", "Htr4", 
       "Edn1", "Egf", "Fgf2",
       "Nrg1", "Nrg2", "Nrg3",
       "Erbb4", "Ngf", "Bdnf", "Ntrk2",
       "Nos1", "Nos2", "Nos3",
       "Il1b", # interleukin 1 beta
       "Inhba", "Calm1", "Calm2", "Calm3",
       "Nrgn", "Calb1", "Calb2", "Gap43", "S100b",
       "Cacna1c", "Cacna1d", "Cacna1s", "Cacna1f",
       "Cacna1b", "Cacna1a", "Cacna1e",
       "Cnga2", "Syp", "Napa",
       "Vamp1", "Vamp2", "Vamp3", "Vamp4", "Vamp5", "Vamp8",
       "Rab3a", "Stx1b", "Syn1", "Snap25",
       "Dlg4", "Rara", "Rarb", "Creb1",
       "Egr1", "Egr2", "Epha5", "Efna5", "Ncam1", "Ncam2",
       "Cdh1", "Cdh2", "Thy1", "Icam5",
       "L1cam", "Ptn", 
        "Itga1", "Itga10","Itga11", "Itga2",  "Itga2b",
       "Itga3",  "Itga4", "Itga5",  "Itga6",  "Itga7",  
       "Itga8",  "Itga9",  "Itgad",  "Itgae", "Itgal",  
       "Itgam",  "Itgav",  "Itgax",  "Itgb1",  "Itgb1bp1",
       "Itgb2",  "Itgb2l", "Itgb3",  "Itgb3bp",  "Itgb4",
       "Itgb5",  "Itgb6", "Itgb7",  "Itgb8",  "Itgbl1",
       "Cd47", "Tnc",
       "Mapk1", "Mapk10", "Mapk11", "Mapk12", "Mapk14", 
       "Mapk3", "Mapk4", "Mapk6", "Mapk7", "Mapk8", "Mapk9",
       "Src", "Fyn", 
       "Prkcg", "Prkg1", "Prkcz", 
       "Capn1", "Capn10", "Capn11", "Capn12", "Capn13",
       "Capn15", "Capn2", "Capn3", "Capn5", "Capn6", 
       "Capn7", "Capn8", "Capn9",
       "Cast", "Serpine2", "Plat", "Plg", "Ube3a",
       "Pla2g10", "Pla2g12a", "Pla2g12b", "Pla2g15",
       "Pla2g16", "Pla2g1b", "Pla2g2a", "Pla2g2c", "Pla2g2d",
       "Pla2g2e", "Pla2g2f", "Pla2g3",  "Pla2g4a", "Pla2g4b",
       "Pla2g4e", "Pla2g4f", "Pla2g5", "Pla2g6", "Pla2g7",
       "Plcb1", "Plcb2", "Plcb3", "Plcb4",
       "Plcg1", "Plcg2",
       "Parp1", "Ppp3ca", "Ppp3cb", "Ppp3cc",
       "Phpt1", "Ache",
       "Adcy1", "Adcy10", "Adcy2", "Adcy3", "Adcy4",
       "Adcy5", "Adcy6", "Adcy7", "Adcy8", "Adcy9",
       "Gucy1a2", "Gucy1a3", "Gucy1b2", "Gucy1b3",
       "Gucy2c", "Gucy2d", "Gucy2e", "Gucy2g",
       "Sptan1", "Sptbn1", "Gfap", "Stmn4",
       "Ccr7", "Mas1",
       "Homer1", "Homer2", "Homer3" )

sanesLichtman[order(sanesLichtman)] # print list alphabetically

# confirm that all all Sanes and Lichtman genes are in the reference transcriptome
sanesLichtman_reference <- geneids %>%
  dplyr::filter(gene %in% sanesLichtman) %>%
  dplyr::select(gene) %>%
  distinct() %>%
  arrange(gene) %>%
  droplevels()
sanesLichtman_reference <- sanesLichtman_reference[,c(1)]
sanesLichtman_reference

# identify which of the Sanes and Lichtman genes are present in my samples
sanesLichtman_present <- dissociation %>%
  dplyr::filter(gene %in% sanesLichtman) %>%
  droplevels()
sanesLichtman_present <- sanesLichtman_present[,c(1)]
sanesLichtman_present

# identify whichof the Sanes and Lichtman genes are differentially expressed in this analysis
sanesLichtman_DEGs <- DEGs %>%
  dplyr::filter(gene %in% sanesLichtman) %>%
  arrange(gene)
sanesLichtman_DEGs

candidates <- dissociation %>%
  dplyr::filter(grepl('Mapk', gene)) %>%
  dplyr::select(gene) %>%
  distinct() %>%
  arrange(gene) %>%
  droplevels()
candidates <- candidates[,c(1)]
candidates

# percent DEGs in the sanes lichtman list
round(6/243*100,2)
round(6/178*100,2)

# Grin2a NMDAR 2A
# Il1b   Interleukin 1 Beta
# Itga5  Integrin Subunit Alpha 5
# Itgam  Integrin Subunit Alpha M
# Itgb5  Integrin Subunit Beta 5
# Mapk3  MAP Kinase 3

```