---
title: "R Notebook"
output: md_document
---

Cho et al 2015 used RNA sequencing to quantify transcript levels in the mouse hippocampus after contextual fear conditioning. The Cho dataset provides a snapshot of gene expression changes associated with hippocampal learning and memory 30 min and 4 h after an experiment. The Cho data are available at <http://science.sciencemag.org/content/suppl/2015/09/30/350.6256.82.DC1> and <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72064>. The file `../data/aac7368-Cho.SM.Table.S2.xls` of differentially expressed genes was used as a representative dataset for learning and memory associated gene expression patterns. 

In this analysis, I compared the Cho et al differentially expressed genes (DEGs) to my experimental results (referred to as the Harris data) to identify the interaction between genes that are differentially expressed following fear condition and following a chemical manipulation.

This analysis prints the list of genes that are differentially expressed in both experiments. The images show that only a few (red dots) of genes that respond to chemical dissociation are up-regulated or down-regulated following fear-conditioning.

[Click here to view the source code.](./08-genelists.Rmd)


```{r, message = F, warning=F, echo = F}
library(readxl)
library(ggplot2)
library(cowplot)
library(dplyr)
library(ggrepel)
library(plyr)


# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/08-genelists/', echo = F, message = F, warning=F)

```



```{r}

### functions for data viz and wrangling

plotvolcano <- function(df, lfc, log10p, plottitle){
  ggplot(df, aes(x = lfc, y = log10p)) + 
  geom_point(aes(color = factor(direction)), size = 1, alpha = 0.8, na.rm = T) + 
  scale_x_continuous(name="log fold change") +
  scale_y_continuous(name="-log10 p-value",
                     breaks = c(0,5,10,15),
                     limits= c(-1,16)) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 ) +
  theme( legend.title = element_blank(),
         legend.position = "bottom",
         legend.text=element_text(size=7))  +
  scale_color_manual(values = c("none" = "grey",
                                "fear-conditioned" = "#018571",
                                "control" = "#a6611a")) + 
  labs(title = plottitle)
}



plotboxplot <- function(df, log10p, direction, plottitle){
  ggplot(df, aes(x = direction, y = log10p, colour = direction)) + 
  geom_boxplot() +
  theme( legend.title = element_blank(),
         legend.position = "bottom")  +
  scale_x_discrete(name="Upregulated in") +
  scale_y_continuous(name="-log10 p-value") +
  labs(title = plottitle) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 )
} 
  
# format Cho data for plotting and data frame joining
wrangleCho <- function(df){
  data <- df  
  data$log10p <- -log10(data$pvalue) 
  data <- data %>% select(gene, lfc, log10p, pvalue, Description) %>%
    mutate(direction = ifelse(data$lfc > 0 & data$log10p > 1, 
                        yes = "fear-conditioned", 
                        no = ifelse(data$lfc < 0 & data$log10p > 1, 
                                    yes = "control", 
                                    no = "none")))
  data$direction <- as.factor(data$direction)
  data$direction <- factor(data$direction, c("control", "none", "fear-conditioned"))
  return(data)
}


top100overlap <- function(Chodataframe){
  df <- Chodataframe %>%
  filter(direction != "none")  %>%
  arrange(lfc) %>%
  tail(n=100)

  ij <- inner_join(df, dissociation, by = "gene")
  names(ij)[6] <- "Cho"
  names(ij)[10] <- "Harris"
  ij <-  ij %>% select(gene, Cho, Harris, log10p, Description) 
  
  return(ij)
}

# overlap with significance for each experiment
overlap <- function(Chodataframe){
  df <- Chodataframe %>%
  #filter(direction != "none")  %>%
  arrange(lfc) 
  #tail(n=100)

  ij <- full_join(df, dissociation, by = "gene")
  names(ij)[6] <- "Cho"
  names(ij)[10] <- "Harris"
  #ij <-  ij %>% select(gene, Cho, Harris, log10p, lfc.x, Description) 
  ij$Harris <- as.character(ij$Harris)
  ij$Harris[is.na(ij$Harris)] <- "absent"
  #ij$Harris <- as.factor(ij$Harris)
  ij$Cho <- as.character(ij$Cho)
  ij$Cho[is.na(ij$Cho)] <- "absent"
  #ij$Cho <- as.factor(ij$Cho)
  
  ij$color <- ifelse(grepl("absent", ij$Harris), ij$Cho, 
                     ifelse(grepl("none", ij$Harris), ij$Cho,
                         ifelse(grepl("HOMO", ij$Harris), "HOMO",
                                ifelse(grepl("DISS", ij$Harris), "DISS",
                                       NA))))
  ij$color <- as.factor(ij$color) 
  ij$color <- factor(ij$color, levels = c( "absent" , "none", 
                                         "control","fear-conditioned",
                                         "HOMO" , "DISS" ))
 #names(useful)
  
  return(ij)
}


## suzy volcano
suzyvolcano <- function(df, lfc, log10p, plottitle){
  ggplot(df, aes(x = lfc, y = log10p)) + 
  geom_point(aes(color = factor(color), size = factor(color)), 
             alpha = 0.8, na.rm = T) +
  scale_size_manual(values=c(0.5, 0.5, 0.5, 0.5, 1, 1)) +
  scale_x_continuous(name="log fold change") +
  scale_y_continuous(name="-log10 p-value",
                     breaks = c(0,5,10,15),
                     limits= c(-1,16)) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 ) +
  theme( legend.title = element_blank(),
         legend.position = "bottom",
         legend.text=element_text(size=7))  +
  scale_color_manual(values = c("none" = "grey",
                                "absent" = "grey",
                                "HOMO" = "black",
                                "DISS" = "red",
                                "fear-conditioned" = "#018571",
                                "control" = "#a6611a")) + 
  labs(title = plottitle)
}

```




```{r dissociation}

### Harris et al. data wranling
dissociation <- read.csv("../results/01_dissociation_volcanoTreatment.csv", 
                         header = T, row.names = 1)
names(dissociation)[5] <- "direction"
dissociation$direction <- factor(dissociation$direction, c("HOMO", "none", "DISS"))
summary(dissociation$direction)

# differential gene expression in the harris data set
dissociationDEGs <- dissociation %>% filter(direction != "none") 
```



```{r, fourhours}

### Cho et al. data at 4 hours 

S2 <- as.data.frame(readxl::read_excel("../data/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

# RNA at 4 hours
fourhoursRNA <- rename(S2, c(`RNA fold change (4 h/control), log2` ="lfc", 
                   `p-value (4 h)` = "pvalue",
                   `Gene Symbol` = "gene"))
fourhoursRNA <- wrangleCho(fourhoursRNA)
summary(fourhoursRNA$direction)

volcanoplot1 <- plotvolcano(fourhoursRNA, fourhoursRNA$lfc, fourhoursRNA$log10p, 
                            plottitle = "Cho DEGs - 4 h")

```




```{r, thirtymin}

### Cho et al. data at 30 min 

S2 <- as.data.frame(readxl::read_excel("../data/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

## RNAseq at 30 min
thirtyminRNA <- rename(S2, c(`RNA fold change (30 min/control), log2` ="lfc", 
                   `p-value (30 min)` = "pvalue",
                   `Gene Symbol` = "gene"))

thirtyminRNA <- wrangleCho(thirtyminRNA)
#summary(thirtyminRNA$direction)

volcanoplot3 <- plotvolcano(thirtyminRNA, thirtyminRNA$lfc, thirtyminRNA$log10p, 
                            plottitle = "Cho DEGs - 30 min")


#plot_grid(volcanoplot1, volcanoplot3)

```




```{r suzyvolcanos}

### Plotting their data with my differential exprssion

overlap30min <- overlap(thirtyminRNA)
overlap4h <- overlap(fourhoursRNA)

overlap30min %>%
  filter(Cho == "control",
         Harris =="DISS")  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

overlap30min %>%
  filter(Cho == "fear-conditioned",
         Harris =="DISS")  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

overlap4h %>%
  filter(Cho == "control",
         Harris =="DISS")  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

overlap4h %>%
  filter(Cho == "fear-conditioned",
         Harris =="DISS" )  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

str(overlap30min)
suzyvolcano1 <- suzyvolcano(overlap30min, overlap30min$lfc.x,overlap30min$log10p, 
                            plottitle = "Cho 30 min and Harris DEGs")



suzyvolcano2 <- suzyvolcano(overlap4h, overlap4h$lfc.x, overlap4h$log10p, 
                            plottitle = "Cho 4 hr and Harris DEGs")


```


```{r overlap}

# side by side plots 
plot_grid(suzyvolcano2, suzyvolcano1, nrow = 1)

```




