---
title: "R Notebook"
output: md_document
---

This script creates a top 100 most differntially expreseed genes from  Cho et al. (`../data/aac7368-Cho.SM.Table.S2.xls` from Andre Fenton) and asks if any are significantlly different in from my dissociation experiment.


```{r, message = F, warning=F}
library(readxl)
library(ggplot2)
library(cowplot)
library(dplyr)
library(ggrepel)

# set output file for figures 
knitr::opts_chunk$set(fig.path = './figures/08-genelists/')

```

### functions for data viz

```{r}

plotvolcano <- function(df, lfc, log10p, plottitle){
  ggplot(df, aes(x = lfc, y = log10p)) + 
  geom_point(aes(color = factor(direction), 
                 shape = factor(direction)), 
             size = 1, alpha = 0.8, na.rm = T) + 
  scale_x_continuous(name="log fold change") +
  scale_y_continuous(name="-log10 p-value") +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 ) +
  theme( legend.title = element_blank(),
         legend.position = "bottom")  +
  labs(title = plottitle) 
}


plotboxplot <- function(df, log10p, direction, plottitle){
  ggplot(df, aes(x = direction, y = log10p, colour = direction)) + 
  geom_boxplot() +
  theme( legend.title = element_blank(),
         legend.position = "bottom")  +
  scale_x_discrete(name="Upregulated in") +
  scale_y_continuous(name="-log10 p-value") +
  labs(title = plottitle) 
} 
  
```


### Harris et al. data

```{r}
dissociation <- read.csv("../results/01_dissociation_volcanoTreatment.csv", header = T, row.names = 1)
names(dissociation)[5] <- "direction"
names(dissociation)

summary(dissociation$direction)
dissociation$direction <- factor(dissociation$direction, c("DISS", "none", "HOMO"))


myboxplot <- plotboxplot(dissociation, dissociation$pvalue, 
                         plottitle = "Harris et. al DEGs")

volcanoplot <- plotvolcano(dissociation, dissociation$lfc, dissociation$pvalue, plottitle = "Harris et. al DEGs")

plot_grid(volcanoplot, myboxplot)

# differential gene expression in the harris data set
dissociation <- dissociation %>%
  filter(direction != "none") %>%
  arrange((padj))

```

### Cho et al. data at 4 hours 

```{r}
S2 <- as.data.frame(readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

# the data for the 4 hour timepoint are in columns 10 and 14. ill rename them for easy
names(S2)[10] <- "lfc"      
names(S2)[14] <- "pvalue"   
names(S2)[2] <- "gene"     # critical for joining dataframes later

S2$log10p <- -log10(S2$pvalue) 

data <- S2 %>% select(gene, lfc, log10p, pvalue, Description)

data <- data %>%
  mutate(direction = ifelse(data$lfc > 0 & data$log10p > 1, 
                        yes = "higher", 
                        no = ifelse(data$lfc < 0 & data$log10p > 1, 
                                    yes = "lower", 
                                    no = "nodifferent")))
data$direction <- as.factor(data$direction)
data$direction <- factor(data$direction, c("lower", "nodifferent", "higher"))


# show the number of DEGs at the 4 hour time point 
summary(data$direction)

volcanoplot <- plotvolcano(data, data$lfc, data$log10p, plottitle = "Cho et. al DEGs at 4 hours")

myboxplot <- plotboxplot(df = data, data$log10p, data$direction, "Cho et. al DEGs at 4 hours")

plot_grid(volcanoplot, myboxplot)
```

### Overlaping DEGs between Harris et al. and Cho et al at 4 hours post treatment

```{r}
data <- data %>%
  filter(direction != "nodifferent")  %>%
  arrange(lfc)

# show the overlap between the Harris and Cho DEGs at 4 hours
ij <- inner_join(data, dissociation, by = "gene")
names(ij)

names(ij)[6] <- "Cho"
names(ij)[10] <- "Harris"

ij %>% select(gene, Cho, Harris, log10p) %>%
  arrange(Cho, Harris,log10p)

```

### Cho et al. data at 30 min 

```{r, thirtymin}
# 30 min 
S2 <- as.data.frame(readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))
#names(S2)

names(S2)[9] <- "lfc"      
names(S2)[13] <- "pvalue"   
names(S2)[2] <- "gene"     # critical for joining dataframes later

S2$log10p <- -log10(S2$pvalue) 

data <- S2 %>% select(gene, lfc, log10p, pvalue, Description)

data <- data %>%
  mutate(direction = ifelse(data$lfc > 0 & data$log10p > 1, 
                        yes = "higher", 
                        no = ifelse(data$lfc < 0 & data$log10p > 1, 
                                    yes = "lower", 
                                    no = "nodifferent")))
data$direction <- as.factor(data$direction)
data$direction <- factor(data$direction, c("lower", "nodifferent", "higher"))

# show the number of DEGs at the 30 min time point 
summary(data$direction)

volcanoplot <- plotvolcano(data, data$lfc, data$log10p, plottitle = "Cho et. al DEGs 30 min post stimulust")

myboxplot <- plotboxplot(df = data, data$log10p, data$direction, "Cho et. al DEGs at 30 min post stimulust")

plot_grid(volcanoplot, myboxplot)

```


### Overlaping DEGs between Harris et al. and Cho et al at 30 min post treatment

```{r}

data <- data %>%
  filter(direction != "nodifferent")  %>%
  arrange(lfc)

# show the overlap between the Harris and Cho DEGs at 30 min
ij <- inner_join(data, dissociation, by = "gene")

names(ij)[6] <- "Cho"
names(ij)[10] <- "Harris"


ij %>% select(gene, Cho, Harris, log10p) %>%
  arrange(Cho, Harris,log10p)
```



```{r, include=F}


candidates <- S2 %>%
  dplyr::filter(
    grepl('Otx2|Cldn2|Col8a1|Clc6|Nrsn1|Ttr|Sostdc1|Enpp2|Kcnj13|Esr|AMPA|Creb1|Otx|Gdf2|Pkd|Grin|Grim|Gria', 
          gene)) %>%
  droplevels()
candidates 

```

### Details

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72064

Genome-wide profilings of transcriptome and translatome in mouse hippocampi after contextual fear conditioning

Memory stabilization after learning requires transcriptional and translational regulations in the brain, yet the temporal molecular changes following learning have not been explored at the genomic scale. We here employed ribosome profiling and RNA sequencing to quantify the translational status and transcript levels in mouse hippocampus following contextual fear conditioning. We identified 104 genes that are dynamically regulated. Intriguingly, our analysis revealed novel repressive regulations in the hippocampus: translational suppression of ribosomal protein-coding genes at basal state; learning-induced early translational repression of specific genes; and late persistent suppression of a subset of genes via inhibition of ESR1/ERÎ± signaling. Further behavioral analyses revealed that Nrsn1, one of the newly identified genes undergoing rapid translational repression, can act as a memory suppressor gene. This study unveils the yet unappreciated importance of gene repression mechanisms in memory formation.

```{r}
## these are all the avilable data sets form Cho et al. 
## I only used supplementary table S2 
S2 <- readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S2.xls", skip = 1 ) 
S1 <- readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S1.xls", skip = 1 )
S3 <- readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S3.xls", skip = 2 )
S4 <- readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S4.xls", skip = 1 )
S5 <- readxl::read_excel("~/Downloads/aac7368-Cho.SM.Table.S5.xls", skip = 1 )

# from GEO GSE72064
mESRPFRNAseq <- read.csv("~/Downloads/GSE72064_mES_RPF_RNA-seq.csv")
PrimaryCulture <- read.csv("~/Downloads/GSE72064_PrimaryCulture_RPF_RNA-seq.csv")
Tissue <- read.csv("~/Downloads/GSE72064_Tissue_RNA-seq_CFC.csv")
Tissue2 <- read.csv("~/Downloads/GSE72064_Tissue_RPF_CFC.csv")
```