---
title: "R Notebook"
output: md_document
---

This script identifies to  differntially expreseed genes from  Cho et al. (`../data/aac7368-Cho.SM.Table.S2.xls` from Andre Fenton) and asks if there is any overlap in the the differentially expressed genes in their study compared to mine.


```{r, message = F, warning=F, echo = F}
library(readxl)
library(ggplot2)
library(cowplot)
library(dplyr)
library(ggrepel)
library(plyr)


# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/08-genelists/', echo = T, message = F, warning=F)

```

### functions for data viz

```{r}

plotvolcano <- function(df, lfc, log10p, plottitle){
  ggplot(df, aes(x = lfc, y = log10p)) + 
  geom_point(aes(color = factor(direction)), size = 1, alpha = 0.8, na.rm = T) + 
  scale_x_continuous(name="log fold change") +
  scale_y_continuous(name="-log10 p-value",
                     breaks = c(0,5,10,15),
                     limits= c(-1,16)) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 ) +
  theme( legend.title = element_blank(),
         legend.position = "bottom",
         legend.text=element_text(size=7))  +
  scale_color_manual(values = c("none" = "grey",
                                "fear-conditioned" = "#018571",
                                "control" = "#a6611a")) + 
  labs(title = plottitle)
}



plotboxplot <- function(df, log10p, direction, plottitle){
  ggplot(df, aes(x = direction, y = log10p, colour = direction)) + 
  geom_boxplot() +
  theme( legend.title = element_blank(),
         legend.position = "bottom")  +
  scale_x_discrete(name="Upregulated in") +
  scale_y_continuous(name="-log10 p-value") +
  labs(title = plottitle) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 )
} 
  
# format Cho data for plotting and data frame joining
wrangleCho <- function(df){
  data <- df  
  data$log10p <- -log10(data$pvalue) 
  data <- data %>% select(gene, lfc, log10p, pvalue, Description) %>%
    mutate(direction = ifelse(data$lfc > 0 & data$log10p > 1, 
                        yes = "fear-conditioned", 
                        no = ifelse(data$lfc < 0 & data$log10p > 1, 
                                    yes = "control", 
                                    no = "none")))
  data$direction <- as.factor(data$direction)
  data$direction <- factor(data$direction, c("control", "none", "fear-conditioned"))
  return(data)
}


top100overlap <- function(Chodataframe){
  df <- Chodataframe %>%
  filter(direction != "none")  %>%
  arrange(lfc) %>%
  tail(n=100)

  ij <- inner_join(df, dissociation, by = "gene")
  names(ij)[6] <- "Cho"
  names(ij)[10] <- "Harris"
  ij <-  ij %>% select(gene, Cho, Harris, log10p, Description) 
  
  return(ij)
}

# overlap with significance for each experiment
overlap <- function(Chodataframe){
  df <- Chodataframe %>%
  #filter(direction != "none")  %>%
  arrange(lfc) 
  #tail(n=100)

  ij <- full_join(df, dissociation, by = "gene")
  names(ij)[6] <- "Cho"
  names(ij)[10] <- "Harris"
  #ij <-  ij %>% select(gene, Cho, Harris, log10p, lfc.x, Description) 
  ij$Harris <- as.character(ij$Harris)
  ij$Harris[is.na(ij$Harris)] <- "absent"
  #ij$Harris <- as.factor(ij$Harris)
  ij$Cho <- as.character(ij$Cho)
  ij$Cho[is.na(ij$Cho)] <- "absent"
  #ij$Cho <- as.factor(ij$Cho)
  
  ij$color <- ifelse(grepl("absent", ij$Harris), ij$Cho, 
                     ifelse(grepl("none", ij$Harris), ij$Cho,
                         ifelse(grepl("HOMO", ij$Harris), "HOMO",
                                ifelse(grepl("DISS", ij$Harris), "DISS",
                                       NA))))
  ij$color <- as.factor(ij$color) 
  ij$color <- factor(ij$color, levels = c( "absent" , "none", 
                                         "control","fear-conditioned",
                                         "HOMO" , "DISS" ))
 #names(useful)
  
  return(ij)
}


## suzy volcano
suzyvolcano <- function(df, lfc, log10p, plottitle){
  ggplot(df, aes(x = lfc, y = log10p)) + 
  geom_point(aes(color = factor(color), size = factor(color)), 
             alpha = 0.8, na.rm = T) +
  scale_size_manual(values=c(0.5, 0.5, 0.5, 0.5, 1, 1)) +
  scale_x_continuous(name="log fold change") +
  scale_y_continuous(name="-log10 p-value",
                     breaks = c(0,5,10,15),
                     limits= c(-1,16)) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 ) +
  theme( legend.title = element_blank(),
         legend.position = "bottom",
         legend.text=element_text(size=7))  +
  scale_color_manual(values = c("none" = "grey",
                                "absent" = "grey",
                                "HOMO" = "black",
                                "DISS" = "red",
                                "fear-conditioned" = "#018571",
                                "control" = "#a6611a")) + 
  labs(title = plottitle)
}

```


### Harris et al. data

```{r dissociation}
dissociation <- read.csv("../results/01_dissociation_volcanoTreatment.csv", 
                         header = T, row.names = 1)
names(dissociation)[5] <- "direction"
dissociation$direction <- factor(dissociation$direction, c("HOMO", "none", "DISS"))
summary(dissociation$direction)

# differential gene expression in the harris data set
dissociationDEGs <- dissociation %>% filter(direction != "none") 
```

### Cho et al. data at 4 hours 

```{r, fourhours}
S2 <- as.data.frame(readxl::read_excel("../data/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

# RNA at 4 hours
fourhoursRNA <- rename(S2, c(`RNA fold change (4 h/control), log2` ="lfc", 
                   `p-value (4 h)` = "pvalue",
                   `Gene Symbol` = "gene"))
fourhoursRNA <- wrangleCho(fourhoursRNA)
summary(fourhoursRNA$direction)

volcanoplot1 <- plotvolcano(fourhoursRNA, fourhoursRNA$lfc, fourhoursRNA$log10p, 
                            plottitle = "Cho DEGs - 4 h")

```


### Cho et al. data at 30 min 

```{r, thirtymin}
# 30 min 
S2 <- as.data.frame(readxl::read_excel("../data/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

## RNAseq at 30 min
thirtyminRNA <- rename(S2, c(`RNA fold change (30 min/control), log2` ="lfc", 
                   `p-value (30 min)` = "pvalue",
                   `Gene Symbol` = "gene"))

thirtyminRNA <- wrangleCho(thirtyminRNA)
#summary(thirtyminRNA$direction)

volcanoplot3 <- plotvolcano(thirtyminRNA, thirtyminRNA$lfc, thirtyminRNA$log10p, 
                            plottitle = "Cho DEGs - 30 min")


plot_grid(volcanoplot1, volcanoplot3)

```



Plotting their data with my differential exprssion

```{r suzyvolcanos}
overlap30min <- overlap(thirtyminRNA)
overlap4h <- overlap(fourhoursRNA)

overlap30min %>%
  filter(Cho == "control",
         Harris =="DISS")  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

overlap30min %>%
  filter(Cho == "fear-conditioned",
         Harris =="DISS")  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

overlap4h %>%
  filter(Cho == "control",
         Harris =="DISS")  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

overlap4h %>%
  filter(Cho == "fear-conditioned",
         Harris =="DISS" )  %>%
  select(gene, lfc.x, log10p) %>%
  arrange(gene)

candidates <- overlap30min %>%
  dplyr::filter(grepl('Fosb|Junb|Nfkbia|Btg2|Btg2', gene)) %>%
  droplevels()
candidates <- candidates[,c(1)]
candidates

str(overlap30min)
suzyvolcano1 <- suzyvolcano(overlap30min, overlap30min$lfc.x,overlap30min$log10p, 
                            plottitle = "Cho 30 min and Harris DEGs")



suzyvolcano2 <- suzyvolcano(overlap4h, overlap4h$lfc.x, overlap4h$log10p, 
                            plottitle = "Cho 4 hr and Harris DEGs")


```

All together again
```{r overlap}
plot_grid(suzyvolcano2, suzyvolcano1, nrow = 1)

```





### Details

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72064

Genome-wide profilings of transcriptome and translatome in mouse hippocampi after contextual fear conditioning

Memory stabilization after learning requires transcriptional and translational regulations in the brain, yet the temporal molecular changes following learning have not been explored at the genomic scale. We here employed ribosome profiling and RNA sequencing to quantify the translational status and transcript levels in mouse hippocampus following contextual fear conditioning. We identified 104 genes that are dynamically regulated. Intriguingly, our analysis revealed novel repressive regulations in the hippocampus: translational suppression of ribosomal protein-coding genes at basal state; learning-induced early translational repression of specific genes; and late persistent suppression of a subset of genes via inhibition of ESR1/ERÎ± signaling. Further behavioral analyses revealed that Nrsn1, one of the newly identified genes undergoing rapid translational repression, can act as a memory suppressor gene. This study unveils the yet unappreciated importance of gene repression mechanisms in memory formation.
