---
title: "R Notebook"
output: md_document
---

This script identifies to  differntially expreseed genes from  Cho et al. (`../data/aac7368-Cho.SM.Table.S2.xls` from Andre Fenton) and asks if there is any overlap in the the differentially expressed genes in their study compared to mine.


```{r, message = F, warning=F, echo = F}
library(readxl)
library(ggplot2)
library(cowplot)
library(dplyr)
library(ggrepel)
library(plyr)

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/08-genelists/', echo = T, message = F, warning=F)

```

### functions for data viz

```{r}

plotvolcano <- function(df, lfc, log10p, plottitle){
  ggplot(df, aes(x = lfc, y = log10p)) + 
  geom_point(aes(color = factor(direction), 
                 shape = factor(direction)), 
             size = 1, alpha = 0.8, na.rm = T) + 
  scale_x_continuous(name="log fold change") +
  scale_y_continuous(name="-log10 p-value") +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 ) +
  theme( legend.title = element_blank(),
         legend.position = "bottom")  +
  labs(title = plottitle)
}


plotboxplot <- function(df, log10p, direction, plottitle){
  ggplot(df, aes(x = direction, y = log10p, colour = direction)) + 
  geom_boxplot() +
  theme( legend.title = element_blank(),
         legend.position = "bottom")  +
  scale_x_discrete(name="Upregulated in") +
  scale_y_continuous(name="-log10 p-value") +
  labs(title = plottitle) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 )
} 
  
# format Cho data for plotting and data frame joining
wrangleCho <- function(df){
  data <- df  
  data$log10p <- -log10(data$pvalue) 
  data <- data %>% select(gene, lfc, log10p, pvalue, Description) %>%
    mutate(direction = ifelse(data$lfc > 0 & data$log10p > 1, 
                        yes = "fear-conditioned", 
                        no = ifelse(data$lfc < 0 & data$log10p > 1, 
                                    yes = "control", 
                                    no = "none")))
  data$direction <- as.factor(data$direction)
  data$direction <- factor(data$direction, c("control", "none", "fear-conditioned"))
  return(data)
}


top100overlap <- function(Chodataframe){
  df <- Chodataframe %>%
  filter(direction != "none")  %>%
  arrange(lfc) %>%
  tail(n=100)
  
  ij <- inner_join(df, dissociation, by = "gene")
  names(ij)[6] <- "Cho"
  names(ij)[10] <- "Harris"
  ij <-  ij %>% select(gene, Cho, Harris, log10p, Description) 
  
  print(ij)
}

```


### Harris et al. data

```{r dissociation}
dissociation <- read.csv("../results/01_dissociation_volcanoTreatment.csv", 
                         header = T, row.names = 1)
names(dissociation)[5] <- "direction"
dissociation$direction <- factor(dissociation$direction, c("HOMO", "none", "DISS"))
summary(dissociation$direction)

myboxplot <- plotboxplot(dissociation, dissociation$pvalue, 
                         plottitle = "Harris et. al DEGs")

volcanoplot <- plotvolcano(dissociation, dissociation$lfc, 
                           dissociation$pvalue, plottitle = "Harris et. al DEGs")

plot_grid(volcanoplot, myboxplot)

# differential gene expression in the harris data set
dissociation <- dissociation %>% filter(direction != "none") 
```

### Cho et al. data at 4 hours 

```{r, fourhours}
S2 <- as.data.frame(readxl::read_excel("../data/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

fourhours <- rename(S2, c(`RPF fold change (4 h/control), log2` ="lfc", 
                   `p-value (4 h)` = "pvalue",
                   `Gene Symbol` = "gene"))

fourhours <- wrangleCho(fourhours)
head(fourhours)
data <- fourhours

summary(data$direction)

volcanoplot <- plotvolcano(data, data$lfc, data$log10p, plottitle = "Cho et. al DEGs at 4 hours")

myboxplot <- plotboxplot(df = data, data$log10p, data$direction, "Cho et. al DEGs at 4 hours")

plot_grid(volcanoplot, myboxplot)
```

### Overlaping DEGs between Harris et al. and Cho et al at 4 hours post treatment

Top 100 DEGs from Cho et al. 4 hours post contextual fear conditioning versus control.
All DEGs from Harris et al. dissocated versus homogenized control
```{r}
top100overlap(fourhours)
```

### Cho et al. data at 30 min 

```{r, thirtymin}
# 30 min 
S2 <- as.data.frame(readxl::read_excel("../data/aac7368-Cho.SM.Table.S2.xls", skip = 1 ))

# select and rename the columns of interest
thirtymin <- rename(S2, c(`RNA fold change (30 min/control), log2` ="lfc", 
                   `p-value (30 min)` = "pvalue",
                   `Gene Symbol` = "gene"))

thirtymin <- wrangleCho(thirtymin)

data <- thirtymin

head(data)
summary(data$direction)

volcanoplot <- plotvolcano(data, data$lfc, data$log10p, plottitle = "Cho et. al DEGs 30 min")

myboxplot <- plotboxplot(df = data, data$log10p, data$direction, "Cho et. al DEGs at 30 min")

plot_grid(volcanoplot, myboxplot)

```


### Overlaping DEGs between Harris et al. and Cho et al at 30 min post treatment

Top 100 DEGs from Cho et al. 30 min post contextual fear conditioning versus control.
All DEGs from Harris et al. dissocated versus homogenized control
```{r}
top100overlap(thirtymin)
```

### Details

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72064

Genome-wide profilings of transcriptome and translatome in mouse hippocampi after contextual fear conditioning

Memory stabilization after learning requires transcriptional and translational regulations in the brain, yet the temporal molecular changes following learning have not been explored at the genomic scale. We here employed ribosome profiling and RNA sequencing to quantify the translational status and transcript levels in mouse hippocampus following contextual fear conditioning. We identified 104 genes that are dynamically regulated. Intriguingly, our analysis revealed novel repressive regulations in the hippocampus: translational suppression of ribosomal protein-coding genes at basal state; learning-induced early translational repression of specific genes; and late persistent suppression of a subset of genes via inhibition of ESR1/ERÎ± signaling. Further behavioral analyses revealed that Nrsn1, one of the newly identified genes undergoing rapid translational repression, can act as a memory suppressor gene. This study unveils the yet unappreciated importance of gene repression mechanisms in memory formation.
