---
title: "04_Cembrowski.Rmd"
author: "Rayna M Harris"
date: December 16, 2017
output: html_document
---

## Reproducing the Cembrowski et al. study (Part 1)

### Get data

This ([2016 Cembrowski paper](https://elifesciences.org/content/5/e14997#fig1s30)) is very similar to my experiment, so I want to compare the two. Like mine, they compare hippocampal gene expression from dorsal CA1, CA3, and DG sub Subfields. These cells were identifed through fac sorting to isolate genetically labeled CA1 and CA3 pyramical neurons and DG granular cells.

Before beginning, I used the following UNIX commands to get their data.

This data was made available here [open source data](https://www.janelia.org/lab/spruston-lab/resources/source-data-simulation-code-other-resources), but I downloaded it from the GenBank archive using the following commands: 

~~~
mkdir ../data/Cembrowski
cd ../data/Cembrowski
wget 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74985/suppl/GSE74985_gene_exp.diff.gz'
wget 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74985/suppl/GSE74985_genes.fpkm_tracking.gz'
wget 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74985/suppl/GSE74985_genes.read_group_tracking.txt.gz'
wget 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74985/suppl/GSE74985_mergedCount.txt.gz'
gunzip *.gz
gzip GSE74985_genes.fpkm_tracking
cd ../../bin
~~~

The GSE74985_genes.fpkm_tracking file is used to extract the gene names with the corresponding ensembl gene id. The file must be unzipped for use in R, but it must be zipped in order to store it on GitHub. The 4985_mergedCount.txt file is used for gene expression analyis in R.

## Refine data

I got this word from "Open Refine", which is a software tool to clean data before analysis. In order to compare the data to mine directly, I want to reanalyze the cembrowksi data using my workflow. So, I need to remove some samples that don't address my question, and re-arrange the data structure so that it is compatable with my workflow. 

```{r setup, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
library(dplyr)
library(plyr)
library(reshape2)
```

# Summary of the Cembrowski Data

Here is an overview of the gene expression counts file from the Cembrowksi study. Here, the column names contains Ensemble identifies for genes while the row names contain the sample names. 

```{r ImportData, echo=FALSE, message=FALSE, results='hide',comment=FALSE, warning=FALSE}
#read count data 
counts <- read.table("../data/Cembrowski/GSE74985_mergedCount.txt")
head(counts)
```

In these next steps, I use a "geneid" file that I created a while back (see ) to average the transcript counts for each gene and to have a column with gene names instead of Ensembl IDs for reference. I also drop some "non-count" columns and ERCC gene counts.

```{r combinedata}
# contains a file with the gene name and transcript id
geneids <- read.table("../data/Cembrowski/geneids.tsv", header=T)
head(geneids)

## join with geneids so we can look at gene level stuff
counts$gene_id <- row.names(counts)
countbygene <- full_join(geneids, counts)
countbygene <- countbygene %>% 
  filter(gene != "-")
countbygene <- countbygene[-c(1)] ## keep gene name and counts for samples)

## lengthen the dataframe, then wide with gene level sums, then make gene the row name, then round the value to nearest integer
countbygene <- melt(countbygene, id=c("gene")) 
countbygene  <- dcast(countbygene, gene ~ variable, value.var= "value", fun.aggregate=sum)
row.names(countbygene) <- countbygene$gene
countbygene[1] <- NULL
countbygene <- round(countbygene)

# getting ready for DESeq2
countData <- countbygene 
head(countData, 9305)  %>% tail()

```

Now, I use the column names to create a dataframe with columns for sample name, hippocampal subfield, and dorsal-ventral gradient location. I however, rename this last column as "Treatment" for ease of use later. Then, I remove the CA2 and CA4 samples.

```{r createColData}
# extract the sample information
colData <- as.data.frame(colnames(countData))
names(colData)[1] <- "RNAseqID"
colData$region <- sapply(strsplit(as.character(colData$RNAseqID),'\\_'), "[", 1)
colData$location <- sapply(strsplit(as.character(colData$RNAseqID),'\\_'), "[", 2)

# rename variables and columns
colData <- rename(colData, c("region"="Subfield"))
colData <- rename(colData, c("location"="Treatment"))
colData$Subfield <- plyr::revalue(colData$Subfield, c("ca1"="CA1"))
colData$Subfield <- plyr::revalue(colData$Subfield, c("ca3"="CA3"))
colData$Subfield <- plyr::revalue(colData$Subfield, c("dg"="DG"))
colData$Treatment <- plyr::revalue(colData$Treatment, c("d"="dorsal"))
colData$Treatment <- plyr::revalue(colData$Treatment, c("v"="ventral"))

# drop CA2 and CA4 columns from both colData and countData
colData <- colData %>% 
  filter(grepl("DG|CA1|CA3", Subfield))  %>% 
  droplevels() ## subsets data
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols))

# filter lowcounts and NAs
countData[countData < 2] <- 0
countData[is.na(countData)] <- 0

colData$Treatment <- as.factor(colData$Treatment)
colData$Subfield <- as.factor(colData$Subfield)
summary(colData)
table(colData$Treatment, colData$Subfield)
dim(countData)
```

```{r writecsv}
write.csv(colData, "../results/04_cembrowksi_colData.csv", row.names = F)
write.csv(countData, "../results/04_cembrowksi_countData.csv", row.names = T)
```

